<!DOCTYPE html>
<html>

  <head>

  <meta charset="utf-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="viewport" content="width=device-width, initial-scale=1">

  <title>Explore master file downloaded from European Nucleotide Archive</title>
  <meta name="description" content="  Required Python libraries: pandas, numpy">

  <link rel="stylesheet" href="/css/main.css">
  <link rel="stylesheet" href="/css/bootstrap.css">
  <link rel="canonical" href="junseonghwan.github.io/2018/06/24/explore-master-file-downloaded-from-european-nucleotide-archive.html">
  <link rel="alternate" type="application/rss+xml" title="Seong-Hwan Jun" href="junseonghwan.github.io/feed.xml" />

  <script type="text/javascript" async
    src="https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-MML-AM_CHTML">
  </script>
  <!--
  <script src="https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML" type="text/javascript"></script>-->

  <!-- Google tag (gtag.js) -->
<script async src="https://www.googletagmanager.com/gtag/js?id=G-302YLVPMCX"></script>
<script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'G-302YLVPMCX');
</script>

</head>


  <body>

    <header class="site-header">

  <div class="wrapper">

    <a class="site-title" href="/">Seong-Hwan Jun</a>

    <nav class="site-nav">
      <a href="#" class="menu-icon">
        <svg viewBox="0 0 18 15">
          <path fill="#424242" d="M18,1.484c0,0.82-0.665,1.484-1.484,1.484H1.484C0.665,2.969,0,2.304,0,1.484l0,0C0,0.665,0.665,0,1.484,0 h15.031C17.335,0,18,0.665,18,1.484L18,1.484z"/>
          <path fill="#424242" d="M18,7.516C18,8.335,17.335,9,16.516,9H1.484C0.665,9,0,8.335,0,7.516l0,0c0-0.82,0.665-1.484,1.484-1.484 h15.031C17.335,6.031,18,6.696,18,7.516L18,7.516z"/>
          <path fill="#424242" d="M18,13.516C18,14.335,17.335,15,16.516,15H1.484C0.665,15,0,14.335,0,13.516l0,0 c0-0.82,0.665-1.484,1.484-1.484h15.031C17.335,12.031,18,12.696,18,13.516L18,13.516z"/>
        </svg>
      </a>

      <div class="trigger">
        
          
          <a class="page-link" href="/code/">Code</a>
          
        
          
        
          
        
          
          <a class="page-link" href="/research/">Research</a>
          
        
      </div>
    </nav>

  </div>

</header>


    <div class="page-content">
      <div class="wrapper">
        <div class="post">

  <header class="post-header">
    <h1 class="post-title">Explore master file downloaded from European Nucleotide Archive</h1>
    <p class="post-meta">Jun 24, 2018</p>
  </header>

  <article class="post-content">
    <ul>
  <li>Required Python libraries: <code class="language-plaintext highlighter-rouge">pandas, numpy</code></li>
</ul>

<p>This page contains exploratory code written in Jupyter Notebook. For tutorial on Jupyter Notebook go <a href="https://plot.ly/python/ipython-notebook-tutorial/">here</a>.</p>

<p>The file is available for download as notebook <a href="/code/ena_explore.ipynb">here</a>.</p>

<div class="language-python highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="kn">import</span> <span class="n">pandas</span> <span class="k">as</span> <span class="n">pd</span>
<span class="kn">import</span> <span class="n">numpy</span> <span class="k">as</span> <span class="n">np</span>

</code></pre></div></div>

<div class="language-python highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">file_name</span> <span class="o">=</span> <span class="s">"table.txt"</span>

<span class="n">df</span> <span class="o">=</span> <span class="n">pd</span><span class="p">.</span><span class="nf">read_table</span><span class="p">(</span><span class="n">file_name</span><span class="p">,</span> <span class="n">sep</span><span class="o">=</span><span class="s">'</span><span class="se">\t</span><span class="s">'</span><span class="p">)</span>
<span class="nf">print</span><span class="p">(</span><span class="n">df</span><span class="p">.</span><span class="n">columns</span><span class="p">)</span> <span class="c1"># check the column names, we want library_name to be there!
</span>
</code></pre></div></div>

<div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>Index(['study_accession', 'sample_accession', 'secondary_sample_accession',
       'experiment_accession', 'run_accession', 'tax_id', 'scientific_name',
       'instrument_model', 'library_name', 'library_layout', 'fastq_bytes',
       'fastq_ftp'],
      dtype='object')
</code></pre></div></div>

<div class="language-python highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">patient_ids</span> <span class="o">=</span> <span class="n">df</span><span class="p">[</span><span class="s">'library_name'</span><span class="p">].</span><span class="nf">astype</span><span class="p">(</span><span class="nb">str</span><span class="p">).</span><span class="nb">str</span><span class="p">[</span><span class="mi">0</span><span class="p">:</span><span class="mi">6</span><span class="p">]</span>
<span class="n">patients</span> <span class="o">=</span> <span class="nf">set</span><span class="p">(</span><span class="n">patient_ids</span><span class="p">)</span>
<span class="nf">print</span><span class="p">(</span><span class="nf">sorted</span><span class="p">(</span><span class="n">patients</span><span class="p">))</span> 
<span class="c1"># there are 21 patients, but there are supposed to be only 20 patients according to the paper
# KTN609 does not appear in the appendix of the paper: https://www.cell.com/cms/attachment/2119295259/2091819478/mmc1.pdf
</span></code></pre></div></div>

<div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>['KTN102', 'KTN115', 'KTN126', 'KTN129', 'KTN132', 'KTN134', 'KTN147', 'KTN152', 'KTN155', 'KTN206', 'KTN210', 'KTN215', 'KTN302', 'KTN304', 'KTN310', 'KTN316', 'KTN317', 'KTN501', 'KTN609', 'KTN612', 'KTN615']
</code></pre></div></div>

<div class="language-python highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c1"># how many data files are there associated with patient KTN102?
</span><span class="n">patient_id</span> <span class="o">=</span> <span class="s">'KTN102'</span>
<span class="nf">print</span><span class="p">(</span><span class="n">np</span><span class="p">.</span><span class="nf">sum</span><span class="p">(</span><span class="n">patient_ids</span> <span class="o">==</span> <span class="n">patient_id</span><span class="p">))</span> <span class="c1"># there are 511 files associated with this patient
</span>
</code></pre></div></div>

<div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>511
</code></pre></div></div>

<p>For this dataset, there are</p>
<ul>
  <li>bulk data: BLOOD, PRE, MID, POST,</li>
  <li>single cell data: scDNA and scRNA.</li>
</ul>

<p>Let’s extract out the file name patterns.</p>

<div class="language-python highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">df2</span> <span class="o">=</span> <span class="n">df</span><span class="p">.</span><span class="n">loc</span><span class="p">[</span><span class="n">patient_ids</span> <span class="o">==</span> <span class="n">patient_id</span><span class="p">]</span>
<span class="nf">sorted</span><span class="p">(</span><span class="nf">list</span><span class="p">(</span><span class="n">df2</span><span class="p">[</span><span class="s">'library_name'</span><span class="p">]))</span>
</code></pre></div></div>

<div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>['KTN1020',
 'KTN1020cells1',
 'KTN1020cells10',
 'KTN1020cells11',
 'KTN1020cells12',
 'KTN1020cells13',
 'KTN1020cells14',
 'KTN1020cells15',
 'KTN1020cells16',
 'KTN1020cells17',
 'KTN1020cells18',
 'KTN1020cells19',
 'KTN1020cells2',
 'KTN1020cells20',
 'KTN1020cells21',
 'KTN1020cells22',
 'KTN1020cells23',
 'KTN1020cells24',
 'KTN1020cells25',
 'KTN1020cells26',
 'KTN1020cells27',
 'KTN1020cells28',
 'KTN1020cells29',
 'KTN1020cells3',
 'KTN1020cells30',
 'KTN1020cells31',
 'KTN1020cells32',
 'KTN1020cells33',
 'KTN1020cells34',
 'KTN1020cells35',
 'KTN1020cells36',
 'KTN1020cells37',
 'KTN1020cells38',
 'KTN1020cells39',
 'KTN1020cells4',
 'KTN1020cells40',
 'KTN1020cells41',
 'KTN1020cells42',
 'KTN1020cells43',
 'KTN1020cells44',
 'KTN1020cells45',
 'KTN1020cells46',
 'KTN1020cells47',
 'KTN1020cells48',
 'KTN1020cells49',
 'KTN1020cells5',
 'KTN1020cells50',
 'KTN1020cells51',
 'KTN1020cells52',
 'KTN1020cells53',
 'KTN1020cells54',
 'KTN1020cells55',
 'KTN1020cells56',
 'KTN1020cells57',
 'KTN1020cells58',
 'KTN1020cells59',
 'KTN1020cells6',
 'KTN1020cells60',
 'KTN1020cells61',
 'KTN1020cells62',
 'KTN1020cells63',
 'KTN1020cells64',
 'KTN1020cells65',
 'KTN1020cells66',
 'KTN1020cells67',
 'KTN1020cells68',
 'KTN1020cells69',
 'KTN1020cells7',
 'KTN1020cells70',
 'KTN1020cells71',
 'KTN1020cells72',
 'KTN1020cells73',
 'KTN1020cells74',
 'KTN1020cells75',
 'KTN1020cells76',
 'KTN1020cells77',
 'KTN1020cells78',
 'KTN1020cells79',
 'KTN1020cells8',
 'KTN1020cells80',
 'KTN1020cells81',
 'KTN1020cells82',
 'KTN1020cells83',
 'KTN1020cells84',
 'KTN1020cells85',
 'KTN1020cells86',
 'KTN1020cells87',
 'KTN1020cells88',
 'KTN1020cells89',
 'KTN1020cells9',
 'KTN1020cells90',
 'KTN1020cells91',
 'KTN1022',
 'KTN102Blood',
 'KTN102OP',
 'KTN102OPcells1',
 'KTN102OPcells10',
 'KTN102OPcells100',
 'KTN102OPcells101',
 'KTN102OPcells102',
 'KTN102OPcells103',
 'KTN102OPcells104',
 'KTN102OPcells105',
 'KTN102OPcells106',
 'KTN102OPcells107',
 'KTN102OPcells108',
 'KTN102OPcells109',
 'KTN102OPcells11',
 'KTN102OPcells110',
 'KTN102OPcells111',
 'KTN102OPcells112',
 'KTN102OPcells113',
 'KTN102OPcells114',
 'KTN102OPcells115',
 'KTN102OPcells116',
 'KTN102OPcells117',
 'KTN102OPcells118',
 'KTN102OPcells119',
 'KTN102OPcells12',
 'KTN102OPcells120',
 'KTN102OPcells121',
 'KTN102OPcells122',
 'KTN102OPcells123',
 'KTN102OPcells124',
 'KTN102OPcells125',
 'KTN102OPcells126',
 'KTN102OPcells127',
 'KTN102OPcells128',
 'KTN102OPcells129',
 'KTN102OPcells13',
 'KTN102OPcells130',
 'KTN102OPcells131',
 'KTN102OPcells132',
 'KTN102OPcells133',
 'KTN102OPcells134',
 'KTN102OPcells135',
 'KTN102OPcells136',
 'KTN102OPcells137',
 'KTN102OPcells138',
 'KTN102OPcells139',
 'KTN102OPcells14',
 'KTN102OPcells140',
 'KTN102OPcells141',
 'KTN102OPcells142',
 'KTN102OPcells143',
 'KTN102OPcells144',
 'KTN102OPcells145',
 'KTN102OPcells146',
 'KTN102OPcells147',
 'KTN102OPcells148',
 'KTN102OPcells149',
 'KTN102OPcells15',
 'KTN102OPcells150',
 'KTN102OPcells151',
 'KTN102OPcells152',
 'KTN102OPcells153',
 'KTN102OPcells154',
 'KTN102OPcells155',
 'KTN102OPcells156',
 'KTN102OPcells157',
 'KTN102OPcells158',
 'KTN102OPcells159',
 'KTN102OPcells16',
 'KTN102OPcells160',
 'KTN102OPcells161',
 'KTN102OPcells162',
 'KTN102OPcells163',
 'KTN102OPcells164',
 'KTN102OPcells165',
 'KTN102OPcells166',
 'KTN102OPcells167',
 'KTN102OPcells168',
 'KTN102OPcells169',
 'KTN102OPcells17',
 'KTN102OPcells170',
 'KTN102OPcells171',
 'KTN102OPcells172',
 'KTN102OPcells173',
 'KTN102OPcells174',
 'KTN102OPcells175',
 'KTN102OPcells176',
 'KTN102OPcells177',
 'KTN102OPcells178',
 'KTN102OPcells179',
 'KTN102OPcells18',
 'KTN102OPcells180',
 'KTN102OPcells181',
 'KTN102OPcells182',
 'KTN102OPcells183',
 'KTN102OPcells184',
 'KTN102OPcells19',
 'KTN102OPcells2',
 'KTN102OPcells20',
 'KTN102OPcells21',
 'KTN102OPcells22',
 'KTN102OPcells23',
 'KTN102OPcells24',
 'KTN102OPcells25',
 'KTN102OPcells26',
 'KTN102OPcells27',
 'KTN102OPcells28',
 'KTN102OPcells29',
 'KTN102OPcells3',
 'KTN102OPcells30',
 'KTN102OPcells31',
 'KTN102OPcells32',
 'KTN102OPcells33',
 'KTN102OPcells34',
 'KTN102OPcells35',
 'KTN102OPcells36',
 'KTN102OPcells37',
 'KTN102OPcells38',
 'KTN102OPcells39',
 'KTN102OPcells4',
 'KTN102OPcells40',
 'KTN102OPcells41',
 'KTN102OPcells42',
 'KTN102OPcells43',
 'KTN102OPcells44',
 'KTN102OPcells45',
 'KTN102OPcells46',
 'KTN102OPcells47',
 'KTN102OPcells48',
 'KTN102OPcells49',
 'KTN102OPcells5',
 'KTN102OPcells50',
 'KTN102OPcells51',
 'KTN102OPcells52',
 'KTN102OPcells53',
 'KTN102OPcells54',
 'KTN102OPcells55',
 'KTN102OPcells56',
 'KTN102OPcells57',
 'KTN102OPcells58',
 'KTN102OPcells59',
 'KTN102OPcells6',
 'KTN102OPcells60',
 'KTN102OPcells61',
 'KTN102OPcells62',
 'KTN102OPcells63',
 'KTN102OPcells64',
 'KTN102OPcells65',
 'KTN102OPcells66',
 'KTN102OPcells67',
 'KTN102OPcells68',
 'KTN102OPcells69',
 'KTN102OPcells7',
 'KTN102OPcells70',
 'KTN102OPcells71',
 'KTN102OPcells72',
 'KTN102OPcells73',
 'KTN102OPcells74',
 'KTN102OPcells75',
 'KTN102OPcells76',
 'KTN102OPcells77',
 'KTN102OPcells78',
 'KTN102OPcells79',
 'KTN102OPcells8',
 'KTN102OPcells80',
 'KTN102OPcells81',
 'KTN102OPcells82',
 'KTN102OPcells83',
 'KTN102OPcells84',
 'KTN102OPcells85',
 'KTN102OPcells86',
 'KTN102OPcells87',
 'KTN102OPcells88',
 'KTN102OPcells89',
 'KTN102OPcells9',
 'KTN102OPcells90',
 'KTN102OPcells91',
 'KTN102OPcells92',
 'KTN102OPcells93',
 'KTN102OPcells94',
 'KTN102OPcells95',
 'KTN102OPcells96',
 'KTN102OPcells97',
 'KTN102OPcells98',
 'KTN102OPcells99',
 'KTN102_0_10',
 'KTN102_0_10_B2',
 'KTN102_0_11',
 'KTN102_0_11_B2',
 'KTN102_0_12',
 'KTN102_0_12_B2',
 'KTN102_0_13',
 'KTN102_0_14',
 'KTN102_0_14_B2',
 'KTN102_0_15_B2',
 'KTN102_0_16',
 'KTN102_0_16_B2',
 'KTN102_0_17',
 'KTN102_0_17_B2',
 'KTN102_0_18',
 'KTN102_0_18_B2',
 'KTN102_0_19',
 'KTN102_0_19_B2',
 'KTN102_0_1_B2',
 'KTN102_0_2',
 'KTN102_0_20',
 'KTN102_0_20_B2',
 'KTN102_0_21',
 'KTN102_0_22',
 'KTN102_0_22_B2',
 'KTN102_0_23',
 'KTN102_0_23_B2',
 'KTN102_0_24',
 'KTN102_0_24_B2',
 'KTN102_0_25',
 'KTN102_0_25_B2',
 'KTN102_0_26',
 'KTN102_0_26_B2',
 'KTN102_0_27',
 'KTN102_0_27_B2',
 'KTN102_0_28',
 'KTN102_0_28_B2',
 'KTN102_0_29',
 'KTN102_0_29_B2',
 'KTN102_0_2_B2',
 'KTN102_0_3',
 'KTN102_0_30',
 'KTN102_0_30_B2',
 'KTN102_0_31',
 'KTN102_0_31_B2',
 'KTN102_0_32',
 'KTN102_0_32_B2',
 'KTN102_0_33',
 'KTN102_0_33_B2',
 'KTN102_0_34',
 'KTN102_0_34_B2',
 'KTN102_0_35',
 'KTN102_0_35_B2',
 'KTN102_0_36',
 'KTN102_0_36_B2',
 'KTN102_0_37',
 'KTN102_0_37_B2',
 'KTN102_0_38',
 'KTN102_0_38_B2',
 'KTN102_0_39',
 'KTN102_0_39_B2',
 'KTN102_0_4',
 'KTN102_0_40',
 'KTN102_0_40_B2',
 'KTN102_0_41',
 'KTN102_0_41_B2',
 'KTN102_0_42',
 'KTN102_0_43',
 'KTN102_0_43_B2',
 'KTN102_0_44',
 'KTN102_0_44_B2',
 'KTN102_0_45',
 'KTN102_0_45_B2',
 'KTN102_0_46',
 'KTN102_0_46_B2',
 'KTN102_0_47',
 'KTN102_0_47_B2',
 'KTN102_0_48',
 'KTN102_0_48_B2',
 'KTN102_0_4_B2',
 'KTN102_0_5',
 'KTN102_0_5_B2',
 'KTN102_0_6',
 'KTN102_0_6_B2',
 'KTN102_0_7',
 'KTN102_0_7_B2',
 'KTN102_0_8',
 'KTN102_0_8_B2',
 'KTN102_0_9',
 'KTN102_0_9_B2',
 'KTN102_0_Pop',
 'KTN102_2_01',
 'KTN102_2_02',
 'KTN102_2_03',
 'KTN102_2_04',
 'KTN102_2_05',
 'KTN102_2_06',
 'KTN102_2_07',
 'KTN102_2_08',
 'KTN102_2_09',
 'KTN102_2_10',
 'KTN102_2_10_B2',
 'KTN102_2_11',
 'KTN102_2_11_B2',
 'KTN102_2_12',
 'KTN102_2_12_B2',
 'KTN102_2_13',
 'KTN102_2_13_B2',
 'KTN102_2_14',
 'KTN102_2_14_B2',
 'KTN102_2_15',
 'KTN102_2_15_B2',
 'KTN102_2_16',
 'KTN102_2_16_B2',
 'KTN102_2_17',
 'KTN102_2_17_B2',
 'KTN102_2_18',
 'KTN102_2_18_B2',
 'KTN102_2_19',
 'KTN102_2_19_B2',
 'KTN102_2_1_B2',
 'KTN102_2_20',
 'KTN102_2_20_B2',
 'KTN102_2_21',
 'KTN102_2_22',
 'KTN102_2_22_B2',
 'KTN102_2_23',
 'KTN102_2_23_B2',
 'KTN102_2_24',
 'KTN102_2_24_B2',
 'KTN102_2_25',
 'KTN102_2_25_B2',
 'KTN102_2_26',
 'KTN102_2_27',
 'KTN102_2_27_B2',
 'KTN102_2_28',
 'KTN102_2_28_B2',
 'KTN102_2_29',
 'KTN102_2_2_B2',
 'KTN102_2_30',
 'KTN102_2_30_B2',
 'KTN102_2_31',
 'KTN102_2_31_B2',
 'KTN102_2_32',
 'KTN102_2_32_B2',
 'KTN102_2_33',
 'KTN102_2_33_B2',
 'KTN102_2_34',
 'KTN102_2_34_B2',
 'KTN102_2_35',
 'KTN102_2_35_B2',
 'KTN102_2_36',
 'KTN102_2_36_B2',
 'KTN102_2_37',
 'KTN102_2_37_B2',
 'KTN102_2_38',
 'KTN102_2_38_B2',
 'KTN102_2_39',
 'KTN102_2_39_B2',
 'KTN102_2_3_B2',
 'KTN102_2_40',
 'KTN102_2_40_B2',
 'KTN102_2_41',
 'KTN102_2_41_B2',
 'KTN102_2_42',
 'KTN102_2_42_B2',
 'KTN102_2_43',
 'KTN102_2_43_B2',
 'KTN102_2_44',
 'KTN102_2_44_B2',
 'KTN102_2_45',
 'KTN102_2_45_B2',
 'KTN102_2_46',
 'KTN102_2_46_B2',
 'KTN102_2_47',
 'KTN102_2_47_B2',
 'KTN102_2_48_B2',
 'KTN102_2_4_B2',
 'KTN102_2_5_B2',
 'KTN102_2_6_B2',
 'KTN102_2_7_B2',
 'KTN102_2_8_B2',
 'KTN102_2_9_B2',
 'KTN102_2_Pop',
 'KTN102_OP_1',
 'KTN102_OP_10',
 'KTN102_OP_11',
 'KTN102_OP_12',
 'KTN102_OP_13',
 'KTN102_OP_14',
 'KTN102_OP_15',
 'KTN102_OP_16',
 'KTN102_OP_17',
 'KTN102_OP_18',
 'KTN102_OP_19',
 'KTN102_OP_2',
 'KTN102_OP_20',
 'KTN102_OP_21',
 'KTN102_OP_22',
 'KTN102_OP_23',
 'KTN102_OP_24',
 'KTN102_OP_25',
 'KTN102_OP_26',
 'KTN102_OP_27',
 'KTN102_OP_28',
 'KTN102_OP_29',
 'KTN102_OP_30',
 'KTN102_OP_31',
 'KTN102_OP_32',
 'KTN102_OP_33',
 'KTN102_OP_34',
 'KTN102_OP_35',
 'KTN102_OP_36',
 'KTN102_OP_37',
 'KTN102_OP_38',
 'KTN102_OP_39',
 'KTN102_OP_4',
 'KTN102_OP_40',
 'KTN102_OP_41',
 'KTN102_OP_42',
 'KTN102_OP_43',
 'KTN102_OP_44',
 'KTN102_OP_45',
 'KTN102_OP_46',
 'KTN102_OP_47',
 'KTN102_OP_48',
 'KTN102_OP_5',
 'KTN102_OP_6',
 'KTN102_OP_7',
 'KTN102_OP_8',
 'KTN102_OP_9',
 'KTN102_OP_Pop']
</code></pre></div></div>

<p>Based on quick browsing of the values, it seems like the library names have the following patterns:</p>
<ul>
  <li>KTN102Blood, KTN1020, KTN1022, KTN102OP seem to refer to bulk samples
    <ul>
      <li>For KTN1020 and KTN102OP, there seem to be library names with the following suffix ‘cells[0-9]+’. These are either DNA or RNA data for population KTN1020 and KTN102OP.</li>
    </ul>
  </li>
  <li>The other library names have the following pattern: KTN102_[0|2|OP]_[0-9|A-Z|a-Z]+
    <ul>
      <li>Some or all of these files are likely to be single cell data, but we do not know which correspond to RNA and which to DNA samples</li>
      <li>We may be able to use the number of fastq.gz files attached to each library to help identify RNA from DNA samples since RNA samples should be single stranded.</li>
      <li>We may also use the file sizes to our advantage.</li>
    </ul>
  </li>
</ul>

<div class="language-python highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c1"># let's check the data sizes for the files to find the bulk samples
</span><span class="n">ret</span> <span class="o">=</span> <span class="nf">list</span><span class="p">(</span><span class="nf">map</span><span class="p">(</span><span class="k">lambda</span> <span class="n">row</span><span class="p">:</span> <span class="n">np</span><span class="p">.</span><span class="nf">asarray</span><span class="p">(</span><span class="nf">str</span><span class="p">(</span><span class="n">row</span><span class="p">).</span><span class="nf">split</span><span class="p">(</span><span class="s">";"</span><span class="p">),</span> <span class="n">dtype</span><span class="o">=</span><span class="nb">int</span><span class="p">),</span> <span class="nf">list</span><span class="p">(</span><span class="n">df2</span><span class="p">[</span><span class="s">'fastq_bytes'</span><span class="p">])))</span>
<span class="n">ret</span> <span class="o">=</span> <span class="n">np</span><span class="p">.</span><span class="nf">asarray</span><span class="p">(</span><span class="n">ret</span><span class="p">)</span>
<span class="n">ret2</span> <span class="o">=</span> <span class="nf">list</span><span class="p">(</span><span class="nf">map</span><span class="p">(</span><span class="k">lambda</span> <span class="n">row</span><span class="p">:</span> <span class="n">np</span><span class="p">.</span><span class="nf">sum</span><span class="p">(</span><span class="n">row</span><span class="p">)</span><span class="o">/</span><span class="n">np</span><span class="p">.</span><span class="nf">power</span><span class="p">(</span><span class="mi">10</span><span class="p">,</span><span class="mi">9</span><span class="p">),</span> <span class="n">ret</span><span class="p">))</span> <span class="c1"># in GB
</span><span class="n">ret2</span> <span class="o">=</span> <span class="n">np</span><span class="p">.</span><span class="nf">asarray</span><span class="p">(</span><span class="n">ret2</span><span class="p">)</span>
<span class="n">df2</span><span class="p">[</span><span class="n">ret2</span> <span class="o">&gt;</span> <span class="mi">1</span><span class="p">]</span>
</code></pre></div></div>

<div>
<style scoped="">
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>study_accession</th>
      <th>sample_accession</th>
      <th>secondary_sample_accession</th>
      <th>experiment_accession</th>
      <th>run_accession</th>
      <th>tax_id</th>
      <th>scientific_name</th>
      <th>instrument_model</th>
      <th>library_name</th>
      <th>library_layout</th>
      <th>fastq_bytes</th>
      <th>fastq_ftp</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>0</th>
      <td>PRJNA396019</td>
      <td>SAMN07457099</td>
      <td>SRS2412441</td>
      <td>SRX3067795</td>
      <td>SRR5906250</td>
      <td>9606</td>
      <td>Homo sapiens</td>
      <td>Illumina HiSeq 4000</td>
      <td>KTN102Blood</td>
      <td>PAIRED</td>
      <td>46088894;3107017681;3884543372</td>
      <td>ftp.sra.ebi.ac.uk/vol1/fastq/SRR590/000/SRR590...</td>
    </tr>
    <tr>
      <th>1</th>
      <td>PRJNA396019</td>
      <td>SAMN07457098</td>
      <td>SRS2412440</td>
      <td>SRX3067794</td>
      <td>SRR5906251</td>
      <td>9606</td>
      <td>Homo sapiens</td>
      <td>Illumina HiSeq 4000</td>
      <td>KTN102OP</td>
      <td>PAIRED</td>
      <td>55533991;3032757154;3772164832</td>
      <td>ftp.sra.ebi.ac.uk/vol1/fastq/SRR590/001/SRR590...</td>
    </tr>
    <tr>
      <th>2</th>
      <td>PRJNA396019</td>
      <td>SAMN07457097</td>
      <td>SRS2412442</td>
      <td>SRX3067793</td>
      <td>SRR5906252</td>
      <td>9606</td>
      <td>Homo sapiens</td>
      <td>Illumina HiSeq 4000</td>
      <td>KTN1022</td>
      <td>PAIRED</td>
      <td>43518505;2498303748;3015988613</td>
      <td>ftp.sra.ebi.ac.uk/vol1/fastq/SRR590/002/SRR590...</td>
    </tr>
    <tr>
      <th>3</th>
      <td>PRJNA396019</td>
      <td>SAMN07457096</td>
      <td>SRS2412443</td>
      <td>SRX3067792</td>
      <td>SRR5906253</td>
      <td>9606</td>
      <td>Homo sapiens</td>
      <td>Illumina HiSeq 4000</td>
      <td>KTN1020</td>
      <td>PAIRED</td>
      <td>41991292;3179486152;3911231851</td>
      <td>ftp.sra.ebi.ac.uk/vol1/fastq/SRR590/003/SRR590...</td>
    </tr>
  </tbody>
</table>
</div>

<div class="language-python highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">ret2</span><span class="p">[</span><span class="n">ret2</span> <span class="o">&gt;</span> <span class="mi">1</span><span class="p">]</span> <span class="c1"># large files probably correspond to bulk, matching our intuition
</span></code></pre></div></div>

<div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>array([ 7.03764995,  6.86045598,  5.55781087,  7.13270929])
</code></pre></div></div>

<div class="language-python highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c1"># let's check the number of files associated with each row
</span><span class="n">num_files</span> <span class="o">=</span> <span class="n">np</span><span class="p">.</span><span class="nf">array</span><span class="p">(</span><span class="nf">list</span><span class="p">(</span><span class="nf">map</span><span class="p">(</span><span class="k">lambda</span> <span class="n">row</span><span class="p">:</span> <span class="n">row</span><span class="p">.</span><span class="n">shape</span><span class="p">[</span><span class="mi">0</span><span class="p">],</span> <span class="n">ret</span><span class="p">)))</span>
</code></pre></div></div>

<div class="language-python highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">s1</span> <span class="o">=</span> <span class="nf">sorted</span><span class="p">(</span><span class="n">df2</span><span class="p">[</span><span class="n">num_files</span> <span class="o">==</span> <span class="mi">1</span><span class="p">][</span><span class="s">"library_name"</span><span class="p">])</span>
<span class="nf">print</span><span class="p">(</span><span class="nf">len</span><span class="p">(</span><span class="n">s1</span><span class="p">))</span> <span class="c1"># 143
</span><span class="n">s1</span>
</code></pre></div></div>

<div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>143





['KTN102_0_10',
 'KTN102_0_11',
 'KTN102_0_12',
 'KTN102_0_13',
 'KTN102_0_14',
 'KTN102_0_16',
 'KTN102_0_17',
 'KTN102_0_18',
 'KTN102_0_19',
 'KTN102_0_2',
 'KTN102_0_20',
 'KTN102_0_21',
 'KTN102_0_22',
 'KTN102_0_23',
 'KTN102_0_24',
 'KTN102_0_25',
 'KTN102_0_26',
 'KTN102_0_27',
 'KTN102_0_28',
 'KTN102_0_29',
 'KTN102_0_3',
 'KTN102_0_30',
 'KTN102_0_31',
 'KTN102_0_32',
 'KTN102_0_33',
 'KTN102_0_34',
 'KTN102_0_35',
 'KTN102_0_36',
 'KTN102_0_37',
 'KTN102_0_38',
 'KTN102_0_39',
 'KTN102_0_4',
 'KTN102_0_40',
 'KTN102_0_41',
 'KTN102_0_42',
 'KTN102_0_43',
 'KTN102_0_44',
 'KTN102_0_45',
 'KTN102_0_46',
 'KTN102_0_47',
 'KTN102_0_48',
 'KTN102_0_5',
 'KTN102_0_6',
 'KTN102_0_7',
 'KTN102_0_8',
 'KTN102_0_9',
 'KTN102_0_Pop',
 'KTN102_2_01',
 'KTN102_2_02',
 'KTN102_2_03',
 'KTN102_2_04',
 'KTN102_2_05',
 'KTN102_2_06',
 'KTN102_2_07',
 'KTN102_2_08',
 'KTN102_2_09',
 'KTN102_2_10',
 'KTN102_2_11',
 'KTN102_2_12',
 'KTN102_2_13',
 'KTN102_2_14',
 'KTN102_2_15',
 'KTN102_2_16',
 'KTN102_2_17',
 'KTN102_2_18',
 'KTN102_2_19',
 'KTN102_2_20',
 'KTN102_2_21',
 'KTN102_2_22',
 'KTN102_2_23',
 'KTN102_2_24',
 'KTN102_2_25',
 'KTN102_2_26',
 'KTN102_2_27',
 'KTN102_2_28',
 'KTN102_2_29',
 'KTN102_2_30',
 'KTN102_2_31',
 'KTN102_2_32',
 'KTN102_2_33',
 'KTN102_2_34',
 'KTN102_2_35',
 'KTN102_2_36',
 'KTN102_2_37',
 'KTN102_2_38',
 'KTN102_2_39',
 'KTN102_2_40',
 'KTN102_2_41',
 'KTN102_2_42',
 'KTN102_2_43',
 'KTN102_2_44',
 'KTN102_2_45',
 'KTN102_2_46',
 'KTN102_2_47',
 'KTN102_2_Pop',
 'KTN102_OP_1',
 'KTN102_OP_10',
 'KTN102_OP_11',
 'KTN102_OP_12',
 'KTN102_OP_13',
 'KTN102_OP_14',
 'KTN102_OP_15',
 'KTN102_OP_16',
 'KTN102_OP_17',
 'KTN102_OP_18',
 'KTN102_OP_19',
 'KTN102_OP_2',
 'KTN102_OP_20',
 'KTN102_OP_21',
 'KTN102_OP_22',
 'KTN102_OP_23',
 'KTN102_OP_24',
 'KTN102_OP_25',
 'KTN102_OP_26',
 'KTN102_OP_27',
 'KTN102_OP_28',
 'KTN102_OP_29',
 'KTN102_OP_30',
 'KTN102_OP_31',
 'KTN102_OP_32',
 'KTN102_OP_33',
 'KTN102_OP_34',
 'KTN102_OP_35',
 'KTN102_OP_36',
 'KTN102_OP_37',
 'KTN102_OP_38',
 'KTN102_OP_39',
 'KTN102_OP_4',
 'KTN102_OP_40',
 'KTN102_OP_41',
 'KTN102_OP_42',
 'KTN102_OP_43',
 'KTN102_OP_44',
 'KTN102_OP_45',
 'KTN102_OP_46',
 'KTN102_OP_47',
 'KTN102_OP_48',
 'KTN102_OP_5',
 'KTN102_OP_6',
 'KTN102_OP_7',
 'KTN102_OP_8',
 'KTN102_OP_9',
 'KTN102_OP_Pop']
</code></pre></div></div>

<div class="language-python highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">s2</span> <span class="o">=</span> <span class="nf">sorted</span><span class="p">(</span><span class="n">df2</span><span class="p">[</span><span class="n">num_files</span> <span class="o">==</span> <span class="mi">2</span><span class="p">][</span><span class="s">"library_name"</span><span class="p">])</span>
<span class="nf">print</span><span class="p">(</span><span class="nf">len</span><span class="p">(</span><span class="n">s2</span><span class="p">))</span> <span class="c1"># 275
</span><span class="n">s2</span>
</code></pre></div></div>

<div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>275





['KTN1020cells1',
 'KTN1020cells10',
 'KTN1020cells11',
 'KTN1020cells12',
 'KTN1020cells13',
 'KTN1020cells14',
 'KTN1020cells15',
 'KTN1020cells16',
 'KTN1020cells17',
 'KTN1020cells18',
 'KTN1020cells19',
 'KTN1020cells2',
 'KTN1020cells20',
 'KTN1020cells21',
 'KTN1020cells22',
 'KTN1020cells23',
 'KTN1020cells24',
 'KTN1020cells25',
 'KTN1020cells26',
 'KTN1020cells27',
 'KTN1020cells28',
 'KTN1020cells29',
 'KTN1020cells3',
 'KTN1020cells30',
 'KTN1020cells31',
 'KTN1020cells32',
 'KTN1020cells33',
 'KTN1020cells34',
 'KTN1020cells35',
 'KTN1020cells36',
 'KTN1020cells37',
 'KTN1020cells38',
 'KTN1020cells39',
 'KTN1020cells4',
 'KTN1020cells40',
 'KTN1020cells41',
 'KTN1020cells42',
 'KTN1020cells43',
 'KTN1020cells44',
 'KTN1020cells45',
 'KTN1020cells46',
 'KTN1020cells47',
 'KTN1020cells48',
 'KTN1020cells49',
 'KTN1020cells5',
 'KTN1020cells50',
 'KTN1020cells51',
 'KTN1020cells52',
 'KTN1020cells53',
 'KTN1020cells54',
 'KTN1020cells55',
 'KTN1020cells56',
 'KTN1020cells57',
 'KTN1020cells58',
 'KTN1020cells59',
 'KTN1020cells6',
 'KTN1020cells60',
 'KTN1020cells61',
 'KTN1020cells62',
 'KTN1020cells63',
 'KTN1020cells64',
 'KTN1020cells65',
 'KTN1020cells66',
 'KTN1020cells67',
 'KTN1020cells68',
 'KTN1020cells69',
 'KTN1020cells7',
 'KTN1020cells70',
 'KTN1020cells71',
 'KTN1020cells72',
 'KTN1020cells73',
 'KTN1020cells74',
 'KTN1020cells75',
 'KTN1020cells76',
 'KTN1020cells77',
 'KTN1020cells78',
 'KTN1020cells79',
 'KTN1020cells8',
 'KTN1020cells80',
 'KTN1020cells81',
 'KTN1020cells82',
 'KTN1020cells83',
 'KTN1020cells84',
 'KTN1020cells85',
 'KTN1020cells86',
 'KTN1020cells87',
 'KTN1020cells88',
 'KTN1020cells89',
 'KTN1020cells9',
 'KTN1020cells90',
 'KTN1020cells91',
 'KTN102OPcells1',
 'KTN102OPcells10',
 'KTN102OPcells100',
 'KTN102OPcells101',
 'KTN102OPcells102',
 'KTN102OPcells103',
 'KTN102OPcells104',
 'KTN102OPcells105',
 'KTN102OPcells106',
 'KTN102OPcells107',
 'KTN102OPcells108',
 'KTN102OPcells109',
 'KTN102OPcells11',
 'KTN102OPcells110',
 'KTN102OPcells111',
 'KTN102OPcells112',
 'KTN102OPcells113',
 'KTN102OPcells114',
 'KTN102OPcells115',
 'KTN102OPcells116',
 'KTN102OPcells117',
 'KTN102OPcells118',
 'KTN102OPcells119',
 'KTN102OPcells12',
 'KTN102OPcells120',
 'KTN102OPcells121',
 'KTN102OPcells122',
 'KTN102OPcells123',
 'KTN102OPcells124',
 'KTN102OPcells125',
 'KTN102OPcells126',
 'KTN102OPcells127',
 'KTN102OPcells128',
 'KTN102OPcells129',
 'KTN102OPcells13',
 'KTN102OPcells130',
 'KTN102OPcells131',
 'KTN102OPcells132',
 'KTN102OPcells133',
 'KTN102OPcells134',
 'KTN102OPcells135',
 'KTN102OPcells136',
 'KTN102OPcells137',
 'KTN102OPcells138',
 'KTN102OPcells139',
 'KTN102OPcells14',
 'KTN102OPcells140',
 'KTN102OPcells141',
 'KTN102OPcells142',
 'KTN102OPcells143',
 'KTN102OPcells144',
 'KTN102OPcells145',
 'KTN102OPcells146',
 'KTN102OPcells147',
 'KTN102OPcells148',
 'KTN102OPcells149',
 'KTN102OPcells15',
 'KTN102OPcells150',
 'KTN102OPcells151',
 'KTN102OPcells152',
 'KTN102OPcells153',
 'KTN102OPcells154',
 'KTN102OPcells155',
 'KTN102OPcells156',
 'KTN102OPcells157',
 'KTN102OPcells158',
 'KTN102OPcells159',
 'KTN102OPcells16',
 'KTN102OPcells160',
 'KTN102OPcells161',
 'KTN102OPcells162',
 'KTN102OPcells163',
 'KTN102OPcells164',
 'KTN102OPcells165',
 'KTN102OPcells166',
 'KTN102OPcells167',
 'KTN102OPcells168',
 'KTN102OPcells169',
 'KTN102OPcells17',
 'KTN102OPcells170',
 'KTN102OPcells171',
 'KTN102OPcells172',
 'KTN102OPcells173',
 'KTN102OPcells174',
 'KTN102OPcells175',
 'KTN102OPcells176',
 'KTN102OPcells177',
 'KTN102OPcells178',
 'KTN102OPcells179',
 'KTN102OPcells18',
 'KTN102OPcells180',
 'KTN102OPcells181',
 'KTN102OPcells182',
 'KTN102OPcells183',
 'KTN102OPcells184',
 'KTN102OPcells19',
 'KTN102OPcells2',
 'KTN102OPcells20',
 'KTN102OPcells21',
 'KTN102OPcells22',
 'KTN102OPcells23',
 'KTN102OPcells24',
 'KTN102OPcells25',
 'KTN102OPcells26',
 'KTN102OPcells27',
 'KTN102OPcells28',
 'KTN102OPcells29',
 'KTN102OPcells3',
 'KTN102OPcells30',
 'KTN102OPcells31',
 'KTN102OPcells32',
 'KTN102OPcells33',
 'KTN102OPcells34',
 'KTN102OPcells35',
 'KTN102OPcells36',
 'KTN102OPcells37',
 'KTN102OPcells38',
 'KTN102OPcells39',
 'KTN102OPcells4',
 'KTN102OPcells40',
 'KTN102OPcells41',
 'KTN102OPcells42',
 'KTN102OPcells43',
 'KTN102OPcells44',
 'KTN102OPcells45',
 'KTN102OPcells46',
 'KTN102OPcells47',
 'KTN102OPcells48',
 'KTN102OPcells49',
 'KTN102OPcells5',
 'KTN102OPcells50',
 'KTN102OPcells51',
 'KTN102OPcells52',
 'KTN102OPcells53',
 'KTN102OPcells54',
 'KTN102OPcells55',
 'KTN102OPcells56',
 'KTN102OPcells57',
 'KTN102OPcells58',
 'KTN102OPcells59',
 'KTN102OPcells6',
 'KTN102OPcells60',
 'KTN102OPcells61',
 'KTN102OPcells62',
 'KTN102OPcells63',
 'KTN102OPcells64',
 'KTN102OPcells65',
 'KTN102OPcells66',
 'KTN102OPcells67',
 'KTN102OPcells68',
 'KTN102OPcells69',
 'KTN102OPcells7',
 'KTN102OPcells70',
 'KTN102OPcells71',
 'KTN102OPcells72',
 'KTN102OPcells73',
 'KTN102OPcells74',
 'KTN102OPcells75',
 'KTN102OPcells76',
 'KTN102OPcells77',
 'KTN102OPcells78',
 'KTN102OPcells79',
 'KTN102OPcells8',
 'KTN102OPcells80',
 'KTN102OPcells81',
 'KTN102OPcells82',
 'KTN102OPcells83',
 'KTN102OPcells84',
 'KTN102OPcells85',
 'KTN102OPcells86',
 'KTN102OPcells87',
 'KTN102OPcells88',
 'KTN102OPcells89',
 'KTN102OPcells9',
 'KTN102OPcells90',
 'KTN102OPcells91',
 'KTN102OPcells92',
 'KTN102OPcells93',
 'KTN102OPcells94',
 'KTN102OPcells95',
 'KTN102OPcells96',
 'KTN102OPcells97',
 'KTN102OPcells98',
 'KTN102OPcells99']
</code></pre></div></div>

<div class="language-python highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">s3</span> <span class="o">=</span> <span class="nf">sorted</span><span class="p">(</span><span class="n">df2</span><span class="p">[</span><span class="n">num_files</span> <span class="o">==</span> <span class="mi">3</span><span class="p">][</span><span class="s">"library_name"</span><span class="p">])</span>
<span class="nf">print</span><span class="p">(</span><span class="nf">len</span><span class="p">(</span><span class="n">s3</span><span class="p">))</span> <span class="c1"># 93
</span><span class="n">s3</span>
</code></pre></div></div>

<div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>93





['KTN1020',
 'KTN1022',
 'KTN102Blood',
 'KTN102OP',
 'KTN102_0_10_B2',
 'KTN102_0_11_B2',
 'KTN102_0_12_B2',
 'KTN102_0_14_B2',
 'KTN102_0_15_B2',
 'KTN102_0_16_B2',
 'KTN102_0_17_B2',
 'KTN102_0_18_B2',
 'KTN102_0_19_B2',
 'KTN102_0_1_B2',
 'KTN102_0_20_B2',
 'KTN102_0_22_B2',
 'KTN102_0_23_B2',
 'KTN102_0_24_B2',
 'KTN102_0_25_B2',
 'KTN102_0_26_B2',
 'KTN102_0_27_B2',
 'KTN102_0_28_B2',
 'KTN102_0_29_B2',
 'KTN102_0_2_B2',
 'KTN102_0_30_B2',
 'KTN102_0_31_B2',
 'KTN102_0_32_B2',
 'KTN102_0_33_B2',
 'KTN102_0_34_B2',
 'KTN102_0_35_B2',
 'KTN102_0_36_B2',
 'KTN102_0_37_B2',
 'KTN102_0_38_B2',
 'KTN102_0_39_B2',
 'KTN102_0_40_B2',
 'KTN102_0_41_B2',
 'KTN102_0_43_B2',
 'KTN102_0_44_B2',
 'KTN102_0_45_B2',
 'KTN102_0_46_B2',
 'KTN102_0_47_B2',
 'KTN102_0_48_B2',
 'KTN102_0_4_B2',
 'KTN102_0_5_B2',
 'KTN102_0_6_B2',
 'KTN102_0_7_B2',
 'KTN102_0_8_B2',
 'KTN102_0_9_B2',
 'KTN102_2_10_B2',
 'KTN102_2_11_B2',
 'KTN102_2_12_B2',
 'KTN102_2_13_B2',
 'KTN102_2_14_B2',
 'KTN102_2_15_B2',
 'KTN102_2_16_B2',
 'KTN102_2_17_B2',
 'KTN102_2_18_B2',
 'KTN102_2_19_B2',
 'KTN102_2_1_B2',
 'KTN102_2_20_B2',
 'KTN102_2_22_B2',
 'KTN102_2_23_B2',
 'KTN102_2_24_B2',
 'KTN102_2_25_B2',
 'KTN102_2_27_B2',
 'KTN102_2_28_B2',
 'KTN102_2_2_B2',
 'KTN102_2_30_B2',
 'KTN102_2_31_B2',
 'KTN102_2_32_B2',
 'KTN102_2_33_B2',
 'KTN102_2_34_B2',
 'KTN102_2_35_B2',
 'KTN102_2_36_B2',
 'KTN102_2_37_B2',
 'KTN102_2_38_B2',
 'KTN102_2_39_B2',
 'KTN102_2_3_B2',
 'KTN102_2_40_B2',
 'KTN102_2_41_B2',
 'KTN102_2_42_B2',
 'KTN102_2_43_B2',
 'KTN102_2_44_B2',
 'KTN102_2_45_B2',
 'KTN102_2_46_B2',
 'KTN102_2_47_B2',
 'KTN102_2_48_B2',
 'KTN102_2_4_B2',
 'KTN102_2_5_B2',
 'KTN102_2_6_B2',
 'KTN102_2_7_B2',
 'KTN102_2_8_B2',
 'KTN102_2_9_B2']
</code></pre></div></div>

<ul>
  <li>Rows that are associated with single file seem to be for scRNA</li>
  <li>Rows associated with two files seem to be for scDNA</li>
  <li>Rows associated with three files seem to be for bulk and/or single cell data (this will have to be determined by reading the paper) but it’s clear from the size of the files, which ones correspond to bulk data.</li>
</ul>

<h1 id="details-from-the-paper">Details from the paper</h1>

<p>The goal of this paper is to study the resistance to chemotherapy in triple negative breast cancer. To that end, they applied scDNA, scRNA, and bulk exome sequencing to profile longitudinal samples from 20 TNBC patients during chemotherapy.</p>

<p>Deep exome sequencing revealed chemo led to clonal extinction (removal of cancer clone) for 10 patients and 10 patients where clone persisted after chemo. In 8 patients, more detailed study was carried out using scDNA-seq to analyze 900 cells and scRNA-seq to analyze 6,862 cells.</p>

<p>An excerpt from the paper:</p>

<blockquote>
  <p>… single-nucleus RNA-sequencing (SNRS) method. SNRS performs automated imaging and selection of up to 1,800 single nuclei in parallel for <strong>3’ mRNA profiling</strong>. We profiled the transcriptomes of 3,370 single nuclei isolated from two matched longitudinal samples per patient from the four clonal extinction patients (P1, P2, P6, P9).</p>
</blockquote>

<p>From this excerpt, it appears that RNA sequencing produces sequencing from only 3’ end, which supports the suspicion that records associated with a single file are RNA-seq data.</p>

  </article>

</div>

      </div>
    </div>

    <footer class="site-footer">

  <div class="wrapper">

<!--    <h2 class="footer-heading">Seong-Hwan Jun</h2> -->

    <div class="footer-col-wrapper">
      <div class="footer-col  footer-col-3">
        <ul class="contact-list">
          <li>Address:</li>
          <li>265 Crittenden Blvd</li> 
          <li>Box 630</li>
          <li>Rochester, NY 14642</li>
        </ul>
      </div>

      <div class="footer-col  footer-col-3">
        <ul class="social-media-list">
          <!--<li>Email: seonghwan_jun AT urmc DOT rochester DOT edu</li>-->

          
          <li>
            <a href="https://github.com/junseonghwan">
              <span class="icon  icon--github">
                <svg viewBox="0 0 16 16">
                  <path fill="#828282" d="M7.999,0.431c-4.285,0-7.76,3.474-7.76,7.761 c0,3.428,2.223,6.337,5.307,7.363c0.388,0.071,0.53-0.168,0.53-0.374c0-0.184-0.007-0.672-0.01-1.32 c-2.159,0.469-2.614-1.04-2.614-1.04c-0.353-0.896-0.862-1.135-0.862-1.135c-0.705-0.481,0.053-0.472,0.053-0.472 c0.779,0.055,1.189,0.8,1.189,0.8c0.692,1.186,1.816,0.843,2.258,0.645c0.071-0.502,0.271-0.843,0.493-1.037 C4.86,11.425,3.049,10.76,3.049,7.786c0-0.847,0.302-1.54,0.799-2.082C3.768,5.507,3.501,4.718,3.924,3.65 c0,0,0.652-0.209,2.134,0.796C6.677,4.273,7.34,4.187,8,4.184c0.659,0.003,1.323,0.089,1.943,0.261 c1.482-1.004,2.132-0.796,2.132-0.796c0.423,1.068,0.157,1.857,0.077,2.054c0.497,0.542,0.798,1.235,0.798,2.082 c0,2.981-1.814,3.637-3.543,3.829c0.279,0.24,0.527,0.713,0.527,1.437c0,1.037-0.01,1.874-0.01,2.129 c0,0.208,0.14,0.449,0.534,0.373c3.081-1.028,5.302-3.935,5.302-7.362C15.76,3.906,12.285,0.431,7.999,0.431z"/>
                </svg>
              </span>

              <span class="username">junseonghwan</span>
            </a>
          </li>
          

          
        </ul>
      </div>
<!--
      <div class="footer-col  footer-col-3">
        <p class="text"></p>
      </div>
    </div>
-->
  </div>

</footer>


  </body>

</html>
