<!DOCTYPE html>
<html>

  <head>
  <meta charset="utf-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="viewport" content="width=device-width, initial-scale=1">

  <title>Genetics Notes</title>
  <meta name="description" content="  Horizontal/Lateral gene transfer: Movement of genetic material between unicellular or multicellular organisms other than by the vertical transmission of DN...">

  <link rel="stylesheet" href="/css/main.css">
  <link rel="stylesheet" href="/css/bootstrap.css">
  <link rel="canonical" href="junseonghwan.github.io/2017/10/24/genetics-notes.html">
  <link rel="alternate" type="application/rss+xml" title="Seong-Hwan Jun" href="junseonghwan.github.io/feed.xml" />

  

  <script type="text/javascript" async
    src="https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-MML-AM_CHTML">
  </script>
  <!--
  <script src="https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML" type="text/javascript"></script>-->

</head>


  <body>

    <header class="site-header">

  <div class="wrapper">

    <a class="site-title" href="/">Seong-Hwan Jun</a>

    <nav class="site-nav">
      <a href="#" class="menu-icon">
        <svg viewBox="0 0 18 15">
          <path fill="#424242" d="M18,1.484c0,0.82-0.665,1.484-1.484,1.484H1.484C0.665,2.969,0,2.304,0,1.484l0,0C0,0.665,0.665,0,1.484,0 h15.031C17.335,0,18,0.665,18,1.484L18,1.484z"/>
          <path fill="#424242" d="M18,7.516C18,8.335,17.335,9,16.516,9H1.484C0.665,9,0,8.335,0,7.516l0,0c0-0.82,0.665-1.484,1.484-1.484 h15.031C17.335,6.031,18,6.696,18,7.516L18,7.516z"/>
          <path fill="#424242" d="M18,13.516C18,14.335,17.335,15,16.516,15H1.484C0.665,15,0,14.335,0,13.516l0,0 c0-0.82,0.665-1.484,1.484-1.484h15.031C17.335,12.031,18,12.696,18,13.516L18,13.516z"/>
        </svg>
      </a>

      <div class="trigger">
        
          
          <a class="page-link" href="/code/">Code</a>
          
        
          
        
          
        
          
          <a class="page-link" href="/research/">Research</a>
          
        
      </div>
    </nav>

  </div>

</header>


    <div class="page-content">
      <div class="wrapper">
        <div class="post">

  <header class="post-header">
    <h1 class="post-title">Genetics Notes</h1>
    <p class="post-meta">Oct 24, 2017</p>
  </header>

  <article class="post-content">
    <ul>
  <li>Horizontal/Lateral gene transfer: Movement of genetic material between unicellular or multicellular organisms other than by the vertical transmission of DNA from parent to offspring. How does this play a role in cancer?</li>
  <li>Orthology analysis: Finding out whether a pair of homologous genes ar <em>orthologs</em>, stemming from a speciation or <em>paralogs</em>, stemming from a gene duplication.</li>
  <li>Genome anotation:</li>
  <li>Species tree: Basically a geneology of taxa included in the study.</li>
  <li>Gene tree: Evolutionary history of genes included in the study. Gene trees can provide evidence for gene duplication events as well as speciation events. Sequences from different homologs can be included in a gene tree; the subsequent analysis should cluster orthologs, thus demonstrating the evolutionary history of the orthologs.</li>
  <li>Gene duplication events: It is a major mechanism through which new genetic material is generated during molecular evolution. It can be defined as any duplication of a region of DNA that contains a gene. Gene duplication can arise as products of several types of errors in <em>DNA replication and repair</em>  machinery as well as through fortuitious capture by selfish genetic elements. <strong>Common sources of gene duplications include ectopic recombination, retrotransposition event, aneuploidy, polyploidy, and replication slippage.</strong> What are these?</li>
  <li>Speciation events: Lineage splitting event that produces two or more separate species.</li>
</ul>

<h3 id="orthology">Orthology</h3>

<p>When two genes have the same function in different species: if the last common ancestor in the phylogenetic gene tree is a speciation event, then the two genes are orthologs, otherwise, if the LCA corresponds to gene duplication, then the two genes are paralogs.</p>

<p>How do we detect the regions of DNA that corresponds to genes?</p>

<div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>Both MPR-based and PrIME-GEM orthology analysis require a gene tree, which typically has been reconstructed from gene sequence data. However, neither of these methods take the sequences directly into account during the reconciliation analysis, even though sequence data may facilitate dating of gene tree vertices and thereby also classification of these into speciations and duplications. The recently proposed so-called species tree aware–gene tree reconstruction (STA–GTR) methods (Åkerborg et al. 2009; Rasmussen and Kellis 2011; Boussau et al. 2013), which take advantage of a species tree, appear to outperform earlier methods in terms of accuracy. These can be used to obtain more reliable gene trees and, thereby, improve any gene tree-based orthology analysis. As a basis for a STA-GTR method, Åkerborg et al. (2009) presented a probabilistic model, the Duplication-Loss (DL) model with i.i.d. Rates across gene tree edges and sequence evolution (DLRS), that defines a posterior distribution over gene trees and reconciliations for a given multiple sequence alignment (MSA) and a species tree. The DLRS model incorporates submodels for gene tree evolution, sequence evolution and a relaxed molecular clock. This model enables the use of the gene sequences directly in the reconciliation analysis, and consequently, also directly in the orthology analysis. Notice, however, that neither the method of Åkerborg et al. (2009), nor that of Rasmussen and Kellis (2011), provides reconciliations or orthology probabilities.
</code></pre></div></div>

<p>Check what <strong>relaxed molecular clock</strong> means.</p>

<div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>A reconciliation can be viewed as a hypothesis of how a gene tree has evolved inside a species tree.
</code></pre></div></div>

<p>Incongruences between species tree and gene tree can be caused by many phonemona. Does a similar problem occur for cancer as well?</p>

<h3 id="types-of-variation">Types of variation</h3>

<ol>
  <li>Copy number variation.</li>
</ol>

<p>Allele prevalence/frequency.</p>

  </article>

</div>

      </div>
    </div>

    <footer class="site-footer">

  <div class="wrapper">

<!--    <h2 class="footer-heading">Seong-Hwan Jun</h2> -->

    <div class="footer-col-wrapper">
      <div class="footer-col  footer-col-3">
        <ul class="contact-list">
          <li>Address:</li>
          <li>265 Crittenden Blvd</li> 
          <li>Box 630</li>
          <li>Rochester, NY 14642</li>
        </ul>
      </div>

      <div class="footer-col  footer-col-3">
        <ul class="social-media-list">
          <!--<li>Email: seonghwan_jun AT urmc DOT rochester DOT edu</li>-->

          
          <li>
            <a href="https://github.com/junseonghwan">
              <span class="icon  icon--github">
                <svg viewBox="0 0 16 16">
                  <path fill="#828282" d="M7.999,0.431c-4.285,0-7.76,3.474-7.76,7.761 c0,3.428,2.223,6.337,5.307,7.363c0.388,0.071,0.53-0.168,0.53-0.374c0-0.184-0.007-0.672-0.01-1.32 c-2.159,0.469-2.614-1.04-2.614-1.04c-0.353-0.896-0.862-1.135-0.862-1.135c-0.705-0.481,0.053-0.472,0.053-0.472 c0.779,0.055,1.189,0.8,1.189,0.8c0.692,1.186,1.816,0.843,2.258,0.645c0.071-0.502,0.271-0.843,0.493-1.037 C4.86,11.425,3.049,10.76,3.049,7.786c0-0.847,0.302-1.54,0.799-2.082C3.768,5.507,3.501,4.718,3.924,3.65 c0,0,0.652-0.209,2.134,0.796C6.677,4.273,7.34,4.187,8,4.184c0.659,0.003,1.323,0.089,1.943,0.261 c1.482-1.004,2.132-0.796,2.132-0.796c0.423,1.068,0.157,1.857,0.077,2.054c0.497,0.542,0.798,1.235,0.798,2.082 c0,2.981-1.814,3.637-3.543,3.829c0.279,0.24,0.527,0.713,0.527,1.437c0,1.037-0.01,1.874-0.01,2.129 c0,0.208,0.14,0.449,0.534,0.373c3.081-1.028,5.302-3.935,5.302-7.362C15.76,3.906,12.285,0.431,7.999,0.431z"/>
                </svg>
              </span>

              <span class="username">junseonghwan</span>
            </a>
          </li>
          

          
        </ul>
      </div>
<!--
      <div class="footer-col  footer-col-3">
        <p class="text"></p>
      </div>
    </div>
-->
  </div>

</footer>


  </body>

</html>
