<!DOCTYPE html>
<html>

  <head>
  <meta charset="utf-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="viewport" content="width=device-width, initial-scale=1">

  <title>Variant calling</title>
  <meta name="description" content="This post covers calling germline variant and somatic variant from healthy and cancer sample. We will begin by following GATK best practices, in particular, ...">

  <link rel="stylesheet" href="/css/main.css">
  <link rel="stylesheet" href="/css/bootstrap.css">
  <link rel="canonical" href="junseonghwan.github.io/2018/07/07/variant-calling.html">
  <link rel="alternate" type="application/rss+xml" title="Seong-Hwan Jun" href="junseonghwan.github.io/feed.xml" />

  

  <script type="text/javascript" async
    src="https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-MML-AM_CHTML">
  </script>
  <!--
  <script src="https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML" type="text/javascript"></script>-->

</head>


  <body>

    <header class="site-header">

  <div class="wrapper">

    <a class="site-title" href="/">Seong-Hwan Jun</a>

    <nav class="site-nav">
      <a href="#" class="menu-icon">
        <svg viewBox="0 0 18 15">
          <path fill="#424242" d="M18,1.484c0,0.82-0.665,1.484-1.484,1.484H1.484C0.665,2.969,0,2.304,0,1.484l0,0C0,0.665,0.665,0,1.484,0 h15.031C17.335,0,18,0.665,18,1.484L18,1.484z"/>
          <path fill="#424242" d="M18,7.516C18,8.335,17.335,9,16.516,9H1.484C0.665,9,0,8.335,0,7.516l0,0c0-0.82,0.665-1.484,1.484-1.484 h15.031C17.335,6.031,18,6.696,18,7.516L18,7.516z"/>
          <path fill="#424242" d="M18,13.516C18,14.335,17.335,15,16.516,15H1.484C0.665,15,0,14.335,0,13.516l0,0 c0-0.82,0.665-1.484,1.484-1.484h15.031C17.335,12.031,18,12.696,18,13.516L18,13.516z"/>
        </svg>
      </a>

      <div class="trigger">
        
          
          <a class="page-link" href="/code/">Code</a>
          
        
          
        
          
        
          
          <a class="page-link" href="/research/">Research</a>
          
        
      </div>
    </nav>

  </div>

</header>


    <div class="page-content">
      <div class="wrapper">
        <div class="post">

  <header class="post-header">
    <h1 class="post-title">Variant calling</h1>
    <p class="post-meta">Jul 7, 2018</p>
  </header>

  <article class="post-content">
    <p>This post covers calling germline variant and somatic variant from healthy and cancer sample. We will begin by following <a href="https://software.broadinstitute.org/gatk/best-practices/">GATK best practices</a>, in particular, we will be using <code class="highlighter-rouge">HaplotypeCaller</code> and <code class="highlighter-rouge">MuTect2</code>.</p>

<p>A tutorial is available <a href="https://software.broadinstitute.org/gatk/documentation/article?id=11136">here</a>.</p>

<h2 id="haplotypecaller">HaplotypeCaller</h2>

<div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code>java <span class="nt">-jar</span> GenomeAnalysisTK.jar <span class="se">\</span>
     <span class="nt">-R</span> reference.fasta <span class="se">\</span>
     <span class="nt">-T</span> HaplotypeCaller <span class="se">\</span>
     <span class="nt">-I</span> sample.bam <span class="se">\</span>
     <span class="nt">-o</span> germline_snv.vcf
</code></pre></div></div>

<p>Difference between gVCF and VCF can be summarized by the following:</p>

<blockquote>
  <p>The key difference between a regular VCF and a gVCF is that the gVCF has records for all sites, whether there is a variant call there or not. The goal is to have every site represented in the file in order to do joint analysis of a cohort in subsequent steps.</p>
</blockquote>

<p><a href="https://gatkforums.broadinstitute.org/gatk/discussion/4017/what-is-a-gvcf-and-how-is-it-different-from-a-regular-vcf">Source</a>.</p>

<p>We will just be using the VCF format for now.</p>

<p>It took about 4.5 hrs on a single core.</p>

<h2 id="mutect2">MuTect2</h2>
<ul>
  <li>MuTect2 does not allow gVCF as for HaplotypeCaller.</li>
  <li>It requires the tumor and the normal samples to be provided as input. It does not call  germline SNVs.</li>
  <li>It seems like MuTect2 is in a beta stage. Perhaps another tool should be used.</li>
</ul>

<div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code> java <span class="nt">-jar</span> GenomeAnalysisTK.jar <span class="se">\</span>
     <span class="nt">-T</span> MuTect2 <span class="se">\</span>
     <span class="nt">-R</span> reference.fasta <span class="se">\</span>
     <span class="nt">-I</span>:tumor tumor.bam <span class="se">\</span>
     <span class="nt">-I</span>:normal normal.bam <span class="se">\</span>
     <span class="nt">-o</span> somatic_snv.vcf
</code></pre></div></div>

<h2 id="resources">Resources</h2>
<ul>
  <li><a href="https://gatkforums.broadinstitute.org/gatk/discussion/1975/how-can-i-use-parallelism-to-make-gatk-tools-run-faster">Parallelization</a> of GATK tools.</li>
</ul>

  </article>

</div>

      </div>
    </div>

    <footer class="site-footer">

  <div class="wrapper">

<!--    <h2 class="footer-heading">Seong-Hwan Jun</h2> -->

    <div class="footer-col-wrapper">
      <div class="footer-col  footer-col-3">
        <ul class="contact-list">
          <li>Address:</li>
          <li>Seong-Hwan Jun</li>
          <li>1100 Fairview Ave. N S2-140</li>
          <li>Seattle, WA 98109</li> 
        </ul>
      </div>

      <div class="footer-col  footer-col-3">
        <ul class="social-media-list">
          <li>Email: sjun2 AT fredhutch dot org</li>

          
          <li>
            <a href="https://github.com/junseonghwan">
              <span class="icon  icon--github">
                <svg viewBox="0 0 16 16">
                  <path fill="#828282" d="M7.999,0.431c-4.285,0-7.76,3.474-7.76,7.761 c0,3.428,2.223,6.337,5.307,7.363c0.388,0.071,0.53-0.168,0.53-0.374c0-0.184-0.007-0.672-0.01-1.32 c-2.159,0.469-2.614-1.04-2.614-1.04c-0.353-0.896-0.862-1.135-0.862-1.135c-0.705-0.481,0.053-0.472,0.053-0.472 c0.779,0.055,1.189,0.8,1.189,0.8c0.692,1.186,1.816,0.843,2.258,0.645c0.071-0.502,0.271-0.843,0.493-1.037 C4.86,11.425,3.049,10.76,3.049,7.786c0-0.847,0.302-1.54,0.799-2.082C3.768,5.507,3.501,4.718,3.924,3.65 c0,0,0.652-0.209,2.134,0.796C6.677,4.273,7.34,4.187,8,4.184c0.659,0.003,1.323,0.089,1.943,0.261 c1.482-1.004,2.132-0.796,2.132-0.796c0.423,1.068,0.157,1.857,0.077,2.054c0.497,0.542,0.798,1.235,0.798,2.082 c0,2.981-1.814,3.637-3.543,3.829c0.279,0.24,0.527,0.713,0.527,1.437c0,1.037-0.01,1.874-0.01,2.129 c0,0.208,0.14,0.449,0.534,0.373c3.081-1.028,5.302-3.935,5.302-7.362C15.76,3.906,12.285,0.431,7.999,0.431z"/>
                </svg>
              </span>

              <span class="username">junseonghwan</span>
            </a>
          </li>
          

          
        </ul>
      </div>
<!--
      <div class="footer-col  footer-col-3">
        <p class="text"></p>
      </div>
    </div>
-->
  </div>

</footer>


  </body>

</html>
