<!DOCTYPE html>
<html>

  <head>
  <meta charset="utf-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="viewport" content="width=device-width, initial-scale=1">

  <title>Processing raw DNA-seq data</title>
  <meta name="description" content="Introduction">

  <link rel="stylesheet" href="/css/main.css">
  <link rel="stylesheet" href="/css/bootstrap.css">
  <link rel="canonical" href="junseonghwan.github.io/2018/06/11/notes-on-processing-bulk-dna-sequence.html">
  <link rel="alternate" type="application/rss+xml" title="Seong-Hwan Jun" href="junseonghwan.github.io/feed.xml" />

  

  <script type="text/javascript" async
    src="https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-MML-AM_CHTML">
  </script>
  <!--
  <script src="https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML" type="text/javascript"></script>-->

</head>


  <body>

    <header class="site-header">

  <div class="wrapper">

    <a class="site-title" href="/">Seong-Hwan Jun</a>

    <nav class="site-nav">
      <a href="#" class="menu-icon">
        <svg viewBox="0 0 18 15">
          <path fill="#424242" d="M18,1.484c0,0.82-0.665,1.484-1.484,1.484H1.484C0.665,2.969,0,2.304,0,1.484l0,0C0,0.665,0.665,0,1.484,0 h15.031C17.335,0,18,0.665,18,1.484L18,1.484z"/>
          <path fill="#424242" d="M18,7.516C18,8.335,17.335,9,16.516,9H1.484C0.665,9,0,8.335,0,7.516l0,0c0-0.82,0.665-1.484,1.484-1.484 h15.031C17.335,6.031,18,6.696,18,7.516L18,7.516z"/>
          <path fill="#424242" d="M18,13.516C18,14.335,17.335,15,16.516,15H1.484C0.665,15,0,14.335,0,13.516l0,0 c0-0.82,0.665-1.484,1.484-1.484h15.031C17.335,12.031,18,12.696,18,13.516L18,13.516z"/>
        </svg>
      </a>

      <div class="trigger">
        
          
          <a class="page-link" href="/code/">Code</a>
          
        
          
        
          
        
          
          <a class="page-link" href="/research/">Research</a>
          
        
      </div>
    </nav>

  </div>

</header>


    <div class="page-content">
      <div class="wrapper">
        <div class="post">

  <header class="post-header">
    <h1 class="post-title">Processing raw DNA-seq data</h1>
    <p class="post-meta">Jun 11, 2018</p>
  </header>

  <article class="post-content">
    <h2 id="introduction">Introduction</h2>

<p>Processing DNA samples consumes significant amount of time. As a beginner, the task looks even more daunting (speaking from my personal experience). In this post, I have summarized knowledge that I gained from various sources on common pre-processing steps to be carried out in preparation for further analysis.</p>

<h2 id="paired-end-sequencing">Paired-end sequencing</h2>

<p>The paired-end sequencing produces reads from both ends of a DNA fragment (one starting from the 3’ end and the other one starting from the 5’ end). Since DNA fragment is usually longer than the read length, the reads will cover different segments of the fragment, although for short reads, it may be possible for the two reads to overlap.</p>

<p>The main benefit of paired-end sequencing compared to single-end sequencing is that it increases confidence in mapping to the reference genome. The paired-end sequencing usually produces two fastq.gz files, <code class="language-plaintext highlighter-rouge">xxx_1.fastq.gz</code> and <code class="language-plaintext highlighter-rouge">xxx_2.fastq.gz</code>, where <code class="language-plaintext highlighter-rouge">xxx</code> refers to file identifer. These two files should have same number of lines, which can be checked via the following command:</p>
<div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="nb">gunzip</span> <span class="nt">-c</span> xxx_1.fastq.gz | <span class="nb">wc</span> <span class="nt">-l</span>
<span class="nb">gunzip</span> <span class="nt">-c</span> xxx_2.fastq.gz | <span class="nb">wc</span> <span class="nt">-l</span>
</code></pre></div></div>
<p>Note that this may take some time depending on the size of the file. Make sure that the number of lines match in both files. At times, there will be third file, <code class="language-plaintext highlighter-rouge">xxx.fastq.gz</code>, this file seems to contain unpaired reads but some claim that it may contain barcodes. [Source: this <a href="https://www.biostars.org/p/16859/">discussion</a> on biostars.org].</p>

<p><strong>TODO: write a post on sequencing technology and refer to that post here. Discuss about RNA-seq technology. Is paired-end sequencing possible for RNA?</strong></p>

<h2 id="checking-the-quality-of-data">Checking the quality of data</h2>

<p>A first step is to ensure the quality of data, which can be carried out using <a href="https://www.bioinformatics.babraham.ac.uk/projects/fastqc/">fastqc</a>. This program generates an HTML report to help the user identify poor data quality. An example of such report is available <a href="/resources/fastqc_1.html">here</a> and <a href="/resources/fastqc_2.html">here</a>. Here are some notes on how to run <code class="language-plaintext highlighter-rouge">fastqc</code>:</p>
<ul>
  <li>Run a separate fastqc for each file: <code class="language-plaintext highlighter-rouge">fastqc xxx_1.fastq.gz -o output_dir</code>. This will generate the report (in the HTML format) to the <code class="language-plaintext highlighter-rouge">output_dir</code>.</li>
  <li>It took me about 1 hour for 3GB <code class="language-plaintext highlighter-rouge">fastq.gz</code> file on Uppmax cluster (Rackham) and 3 hours for 30GB file on Uppmax cluster (Bianca). Make sure to allocate about 4-6 hours of computing time.</li>
  <li>There doesn’t seem to be any multi-threading support for <code class="language-plaintext highlighter-rouge">fastqc</code> so when requesting computing resources, single core will suffice. Doing so will allow the job to be scheduled faster.</li>
  <li>For details on how to read the report generated, refer to the <a href="https://www.bioinformatics.babraham.ac.uk/projects/fastqc/">documentation</a>.</li>
</ul>

<p><strong>TODO: write a separate post on how to read/interpret the report.</strong></p>

<h2 id="trimming-adapters">Trimming adapters</h2>
<p>The next step is to trim the adapters. The adapters are artefacts of sequencing, where these adapters correspond to the chemical agents that latch on to the fragment. These adapters need to be removed before analysis.</p>

<p><strong>TODO: explain how the adapters arise in another post.</strong></p>

<p>The tool that we will use for carrying out this task is <code class="language-plaintext highlighter-rouge">cutadapt</code>. However, this tool does more than just remove adapters. Some of its capabilities include:</p>
<ol>
  <li>Adapter removal
    <ul>
      <li>This step requires the user to know the adapter sequence. The adapters are platform dependent and can be looked up in the manual.</li>
    </ul>
  </li>
  <li>Quality trimming
    <ul>
      <li>Example: <code class="language-plaintext highlighter-rouge">-q 20</code> will remove the bases with quality score below 20. Similarly, <code class="language-plaintext highlighter-rouge">-q 20,15</code> for paired reads will remove bases with quality score below 20 from 5’ end reads and 15 from 3’ end reads. If only one value is specified but the reads are paired, then only 3’ reads will be processed according to the documentation.</li>
    </ul>
  </li>
  <li>Filtering by read length:
    <ul>
      <li>Q: When filtering reads based on length via <code class="language-plaintext highlighter-rouge">-m l</code>, if a read from 3’ has length shorter than <code class="language-plaintext highlighter-rouge">l</code>, then the corresponding read in 5’ end will also be discarded even though the corresponding read is longer than <code class="language-plaintext highlighter-rouge">l</code>?</li>
      <li><strong>TODO: run it with a very small file and see how it behaves.</strong></li>
    </ul>
  </li>
  <li>The full list of operations is available <a href="http://cutadapt.readthedocs.io/en/stable/guide.html#read-modification-order">here</a>.</li>
</ol>

<h3 id="notes">Notes:</h3>
<ol>
  <li>Many of these operations are performed on both reads; therefore, the two files need to be processed (specified to <code class="language-plaintext highlighter-rouge">cutadapt</code>) together.</li>
  <li>Pipelining the output of <code class="language-plaintext highlighter-rouge">cutadapt</code> to the next step is possible and may be a good idea when storage is a concern. However, it may be of interest to run <code class="language-plaintext highlighter-rouge">fastqc</code> again on the trimmed data just to ensure that everything looks good after trimming.</li>
  <li>Q: <code class="language-plaintext highlighter-rouge">cutadapt</code> is written in Python. Are there better choices for this task?</li>
  <li>There are some discussions on whether this step is even necessary. For example, see this <a href="https://gatkforums.broadinstitute.org/gatk/discussion/9921/trimming-of-adapter-sequence">discussion</a> and <a href="https://www.ecseq.com/support/ngs/trimming-adapter-sequences-is-it-necessary">this</a>).</li>
  <li>This step can be parallelized so request large number of cores. Using 4 cores, it took me about 11 hours to process 20GB file.</li>
</ol>

<h2 id="mapping-to-reference-and-sorting-the-bam-file">Mapping to reference and sorting the BAM file</h2>
<p>For reference mapping task there seem to be two popular choices: <code class="language-plaintext highlighter-rouge">BWA-mem</code> and <code class="language-plaintext highlighter-rouge">bowtie2</code>. I will discuss <code class="language-plaintext highlighter-rouge">bowtie2</code> here. The output of these tools is in SAM (sequence alignment map) file format. A potential problem is that the output file can get rather big and hence, the recommended practice is to pipe the output of <code class="language-plaintext highlighter-rouge">bowtie2</code> to <code class="language-plaintext highlighter-rouge">samtools</code> to produce the BAM (binary alignment map) files.</p>

<p>Note: BAM file format is a compressed version of SAM file format.</p>

<p>Running <code class="language-plaintext highlighter-rouge">bowtie2</code> can be accomlished by issueing the following command:</p>

<div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code>bowtie2 <span class="nt">--maxins</span> 2000 <span class="nt">--threads</span> 16 <span class="nt">-1</span> xxx_1.fastq.gz <span class="nt">-2</span> xxx_2.fastq.gz <span class="nt">-x</span> /path/to/reference <span class="nt">-S</span> <span class="nv">$4</span>
</code></pre></div></div>

<p>Here is description of each of the options:</p>
<ul>
  <li><code class="language-plaintext highlighter-rouge">-1</code>: file name of first of the two paired-end sequencing</li>
  <li><code class="language-plaintext highlighter-rouge">-2</code>: the second of the two files</li>
  <li><code class="language-plaintext highlighter-rouge">-x</code>: the location of the reference genome. Note that these are platform specific. For Uppmax, use the following path: <code class="language-plaintext highlighter-rouge">/sw/data/uppnex/reference/Homo_sapiens/GRCh37/program_files/bowtie2/concat</code></li>
  <li><code class="language-plaintext highlighter-rouge">--minins</code> or <code class="language-plaintext highlighter-rouge">-I</code>: specifies the minimum gap between the location of the first and the second pair. Default value is 0.</li>
  <li><code class="language-plaintext highlighter-rouge">--maxins</code> or <code class="language-plaintext highlighter-rouge">-X</code>: specifies the maximum gap between the location of the first and the second pair. The default value is 500.</li>
  <li><code class="language-plaintext highlighter-rouge">--threads</code> or <code class="language-plaintext highlighter-rouge">-p</code>: number of threads to use</li>
  <li><code class="language-plaintext highlighter-rouge">-S</code>: output file (<strong>but not recommended to output SAM file</strong>)</li>
  <li>Refer to the <a href="http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#the-bowtie2-aligner">manual</a> for complete list of options.</li>
</ul>

<p>Note that larger the gap between <code class="language-plaintext highlighter-rouge">-I</code> and <code class="language-plaintext highlighter-rouge">-X</code>, slower the mapping process. If you have good understanding of the sequencing technology that is used, then specifying accurate values for <code class="language-plaintext highlighter-rouge">-I,-Z</code> will shorten the mapping time.</p>

<p>The recommended tool for compressing to BAM file is <code class="language-plaintext highlighter-rouge">samtools</code>. It is a collection of tools that are frequently used in analyzing SAM/BAM files. Mapping and piping can be accomplished by,</p>

<div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code>bowtie2 <span class="nt">--maxins</span> 2000 <span class="nt">--threads</span> 16 <span class="nt">-1</span> xxx_1.fastq.gz <span class="nt">-2</span> xxx_2.fastq.gz <span class="nt">-x</span> /path/to/reference | samtools view <span class="nt">-h</span> <span class="nt">-b</span> - <span class="o">&gt;</span> mapped.bam
</code></pre></div></div>

<ul>
  <li><code class="language-plaintext highlighter-rouge">view</code>: program name,</li>
  <li><code class="language-plaintext highlighter-rouge">-b</code>: convert to BAM format</li>
  <li><code class="language-plaintext highlighter-rouge">-h</code>: output the header in the BAM file</li>
  <li><code class="language-plaintext highlighter-rouge">-</code>: the dash is telling <code class="language-plaintext highlighter-rouge">samtools</code> that the input is coming from standard input (coming from pipeline).</li>
  <li>It seems like multi-threading is available via <code class="language-plaintext highlighter-rouge">@ num_threads</code> option but some people have reported that it can be unstable.</li>
  <li>The manual is available <a href="http://www.htslib.org/doc/samtools.html">here</a>.</li>
</ul>

<p>Finally, we need to sort the BAM file:</p>

<div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code>samtools <span class="nb">sort</span> <span class="nt">-o</span> sorted.bam mapped.bam
<span class="nb">rm </span>mapped.bam <span class="c"># remove the intermediate BAM file</span>
</code></pre></div></div>

<p><strong>Question: why do we need to sort BAM?</strong></p>
<ul>
  <li>Basically for faster processing down the stream. See this <a href="https://wikis.univ-lille1.fr/bilille/_media/formation_variants_reads_2017.pdf">slide</a>.</li>
</ul>

<p>Some people suggest that writing raw BAM file is also a waste of space (and waste of disk I/O) so, only the sorted file should be stored in the disk. But then, some point out that piping too many commands can also create a bottleneck.</p>

<p><strong>TODO: I have not tried this myself but at some point, try:</strong></p>

<div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code>bowtie2 <span class="nt">--maxins</span> 2000 <span class="nt">--threads</span> 16 <span class="nt">-1</span> xxx_1.fastq.gz <span class="nt">-2</span> xxx_2.fastq.gz <span class="nt">-x</span> /path/to/reference | samtools view <span class="nt">-h</span> <span class="nt">-b</span> - | samtools <span class="nb">sort</span> <span class="nt">-o</span> sorted.bam
</code></pre></div></div>

<p>And also, try with <code class="language-plaintext highlighter-rouge">@16</code> to see if multi-threading makes a difference.</p>

<p>Better yet, it is possible to bypass the step of converting SAM to BAM and just jump straight to sorted BAM:</p>

<div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code>bowtie2 <span class="nt">--maxins</span> 2000 <span class="nt">--threads</span> 16 <span class="nt">-1</span> xxx_1.fastq.gz <span class="nt">-2</span> xxx_2.fastq.gz <span class="nt">-x</span> /path/to/reference | samtools <span class="nb">sort</span> -@8 <span class="nt">-o</span> output.bam -
</code></pre></div></div>

<h3 id="time-comparison">Time comparison</h3>
<p>Compare running times and file outputs of the following scenarios on 3GBx2 files:</p>
<ol>
  <li>Mapping to output SAM file (16 cores). Took about 2hours. Produced 70GB file.</li>
  <li>Converting SAM to BAM file (single-threading) from step 1. <strong>TODO</strong></li>
  <li>Sorting the BAM file (16 cores) from step 2. <strong>TODO</strong></li>
  <li>Piping the output of map to sort directly (16 cores for both map and sort). Took about 2 hours. Produced an 8GB BAM file.</li>
  <li>Piping the output of map to view for unsorted BAM file (use 16 cores for map but single threading for view). Took about 2 hours. Produced an 8GB BAM file.</li>
</ol>

<h2 id="marking-duplicate">Marking duplicate</h2>
<p>The exact reasoning as to why this problem arises is not known to me yet. It appears to be an artefact of sequencing technology.</p>

<p><strong>TODO: read up on the sequencing technologies to try and understand why this happens.</strong></p>

<p>The tool that we will use for marking duplicates is <code class="language-plaintext highlighter-rouge">picard</code>.</p>

<div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c"># requiring 5GB of memory, for Rackham, each core comes with 6.4GB so we should be OK without additional request for memory</span>
java <span class="nt">-Xmx5g</span> <span class="nt">-jar</span> <span class="k">${</span><span class="nv">PICARD</span><span class="k">}</span>/picard.jar <span class="se">\</span>
MarkDuplicates <span class="nv">VALIDATION_STRINGENCY</span><span class="o">=</span>LENIENT <span class="se">\</span>
<span class="nv">INPUT</span><span class="o">=</span>sorted.bam <span class="se">\</span>
<span class="nv">OUTPUT</span><span class="o">=</span>marked.bam.bam <span class="se">\</span>
<span class="nv">METRICS_FILE</span><span class="o">=</span>duplicate_metric.txt 1&gt;&amp;2 2&gt; log_fix_mark.txt
</code></pre></div></div>

<p>This script requires to provide the location of <code class="language-plaintext highlighter-rouge">picard.jar</code> and the input BAM file. The <code class="language-plaintext highlighter-rouge">METRICS_FILE</code> points to the file that will contain the duplication statistics. And the last bit is just to redirect the output to a log file.</p>

<p><strong>TODO: add a bit more on how to read and interpret the metrics file.</strong></p>

<p>The above command wil mark but not remove the duplicates (which is the recommended practice according to GATK <a href="https://gatkforums.broadinstitute.org/gatk/discussion/6747/how-to-mark-duplicates-with-markduplicates-or-markduplicateswithmatecigar">source</a>):</p>

<blockquote>
  <p>By default the tools flag duplicates and retain them in the output file. To remove the duplicate records from the resulting file, set the REMOVE_DUPLICATES parameter to true. However, given you can set GATK tools to include duplicates in analyses by adding -drf DuplicateRead to commands, a better option for value-added storage efficiency is to retain the resulting marked file over the input file.</p>
</blockquote>

<p>The source above states that  we can create the index file at this point:</p>

<blockquote>
  <p>To optionally create a .bai index, add and set the CREATE_INDEX parameter to true.</p>
</blockquote>

<p>Note that Java is given 5GB to work with, noting that one core of Rackham comes with 6.4GB, we do not need to request for additional memory resources.</p>

<p>It took about 1.5 hr to complete this task on Rackham, on the sorted BAM file.</p>

<h2 id="adding-read-group">Adding read group</h2>
<p>This step is needed because GATK does not allow BAM files that do not have read groups added. The command is as follows:</p>
<div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code>java <span class="nt">-Xmx5g</span> <span class="nt">-jar</span> <span class="k">${</span><span class="nv">PICARD</span><span class="k">}</span>/picard.jar <span class="se">\</span>
AddOrReplaceReadGroups <span class="se">\</span>
<span class="nv">INPUT</span><span class="o">=</span>marked.bam <span class="se">\</span>
<span class="nv">OUTPUT</span><span class="o">=</span>fixed.bam <span class="se">\</span>
<span class="nv">CREATE_INDEX</span><span class="o">=</span><span class="nb">true</span> <span class="se">\</span>
<span class="nv">RGID</span><span class="o">=</span><span class="nv">$1</span> <span class="se">\</span>
<span class="nv">RGSM</span><span class="o">=</span><span class="nv">$2</span> <span class="se">\</span>
<span class="nv">RGLB</span><span class="o">=</span><span class="nv">$3</span> <span class="se">\</span>
<span class="nv">RGPL</span><span class="o">=</span>ILLUMINA <span class="se">\</span>
<span class="nv">RGPU</span><span class="o">=</span><span class="s2">"NA"</span> 1&gt;&amp;2 2&gt; log_fix_addreplace.txt
</code></pre></div></div>
<ul>
  <li><code class="language-plaintext highlighter-rouge">CREATE_INDEX</code>: create index file while adding the read group.</li>
  <li><code class="language-plaintext highlighter-rouge">RGID</code>: Read group ID. The default value is 1, and it is recommended to change this to something other than the default value. For example, 2 or if the data is triple negative breast cancer, TNBC or could be a patient ID if there are multiple patients.</li>
  <li><code class="language-plaintext highlighter-rouge">RGSM</code>: Read group sample name. This can be patient ID and/or information on the sample like blood or stomach if such is available.</li>
  <li><code class="language-plaintext highlighter-rouge">RGLB</code>: Read group library (library preparation method?). When unknown just fill in some default value like “NA”.</li>
  <li><code class="language-plaintext highlighter-rouge">RGPL</code>: Read group (sequencing) platform. In my case, we are using ILLUMINA.</li>
  <li><code class="language-plaintext highlighter-rouge">RGPU</code>: Read group platform unit. It is a required filled but I’m not sure what this is supposed to be referring to so, filled in NA.</li>
</ul>

<h3 id="creating-index-file">Creating index file</h3>
<p>Note that alternatively, we can create index file using <code class="language-plaintext highlighter-rouge">samtools index marked.bam marked.bai</code>, this will produce an output <code class="language-plaintext highlighter-rouge">marked.bai</code> (in the order of few MBs for 8GB BAM file).</p>

<h2 id="realignment">Realignment</h2>
<p>First, some notes:</p>
<ul>
  <li>Genome analysis toolkit (GATK) can be used. But this step seems to be optional if using <code class="language-plaintext highlighter-rouge">MuTect2</code> and <code class="language-plaintext highlighter-rouge">HaplotypeCaller</code>.</li>
  <li>Do not do this step for amplicon sequencing (i.e., targeted sequencing data).</li>
  <li>I think it’s case-by-case. If the producer of the data did this step, then do it. Otherwise, it is optional and depends on the variant calling tool that you are using.</li>
</ul>

<p>The pre-req for running indel realignment are the files containing the reference sequence, target interval, and known indel locations. Note that to perform re-alignment over the entire genome would be very time consuming, so it is recommended to carry out local re-alignment around the regions that are known to have insertions and deletions.</p>

<p>The reference sequence and known indel locations are usually available on a large compute clusters like Uppmax. But the target interval file may not available, in which case, we need to create one:</p>

<div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code> java <span class="nt">-Xmx1g</span> <span class="nt">-jar</span> /path/to/GenomeAnalysisTK.jar <span class="se">\</span>
  <span class="nt">-T</span> RealignerTargetCreator <span class="se">\</span>
  <span class="nt">-R</span> /path/to/reference.fasta <span class="se">\</span>
  <span class="nt">-o</span> /path/to/output.intervals <span class="se">\</span>
  <span class="nt">-known</span> /path/to/indel_calls.vcf
</code></pre></div></div>

<p>Then, we call indel realignment as follows:</p>

<div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code>java <span class="nt">-Xmx4g</span> <span class="nt">-Djava</span>.io.tmpdir<span class="o">=</span>/path/to/tmpdir <span class="se">\</span>
  <span class="nt">-jar</span> /path/to/GenomeAnalysisTK.jar <span class="se">\</span>
  <span class="nt">-I</span> fixed.bam <span class="se">\</span>
  <span class="nt">-R</span> &lt;ref.fasta&gt; <span class="se">\</span>
  <span class="nt">-T</span> IndelRealigner <span class="se">\</span>
  <span class="nt">-targetIntervals</span> /path/to/output.intervals <span class="se">\</span>
  <span class="nt">-o</span> realigned.bam <span class="se">\</span>
  <span class="nt">-known</span> /path/to/indel_calls.vcf
</code></pre></div></div>

<p><strong>Note 1</strong> If you don’t add the read groups, you will likely get the following error:</p>

<blockquote>
  <p>Error details: SAM file doesn’t have any read groups defined in the header.  The GATK no longer supports SAM files without read groups.</p>
</blockquote>

<p><strong>Note 2</strong> Error regarding incompatible contigs</p>

<p>It can happen when the names of the chromosome used in the reads differ from the names of the chromosomes used in the reference file. For example, in my case, the reads referred to them as 1, 2, …, 22 but in the reference, they were labelled chr1, chr2, … chr22.</p>

<p>This problem and the possible solutions are laid out in this <a href="https://software.broadinstitute.org/gatk/documentation/article?id=63">post</a>.</p>

<p>Chances are, people in your lab have experienced this issue before and a compatible file format is available somewhere in Uppmax. This is how I solved this issue.</p>

<p>It took about 9.5 hours (single core) on 8GB BAM file.</p>

<h2 id="additional-steps-for-cleaning">Additional steps for cleaning</h2>
<p>These seemed less important to me, so I skipped them. But for a careful and accurate analysis, perhaps these steps should not be skipped:</p>
<ul>
  <li><code class="language-plaintext highlighter-rouge">samtools flagstat</code></li>
  <li>Additional filtering to remove low quality reads, unmapped reads, and marked duplicates</li>
  <li>Base quality score recalibration
    <ul>
      <li>Uses studies on biases of sequencing platform to correct the quality scores. Certain platforms are known to produce more G’s vs A’s for example.</li>
    </ul>
  </li>
</ul>

<p>At this point, the realigned BAM file is more or less ready for analysis such as variant calling. And perhaps even copy number analysis. These next steps will be described in the future posts.</p>

<h2 id="useful-resources">Useful resources:</h2>
<ul>
  <li><a href="https://gatkforums.broadinstitute.org/gatk/discussion/2909#addRG">Adding read grouop</a></li>
  <li><a href="https://gatkforums.broadinstitute.org/gatk/discussion/6747/how-to-mark-duplicates-with-markduplicates-or-markduplicateswithmatecigar">Marking duplicate</a></li>
</ul>

  </article>

</div>

      </div>
    </div>

    <footer class="site-footer">

  <div class="wrapper">

<!--    <h2 class="footer-heading">Seong-Hwan Jun</h2> -->

    <div class="footer-col-wrapper">
      <div class="footer-col  footer-col-3">
        <ul class="contact-list">
          <li>Address:</li>
          <li>265 Crittenden Blvd</li> 
          <li>Box 630</li>
          <li>Rochester, NY 14642</li>
        </ul>
      </div>

      <div class="footer-col  footer-col-3">
        <ul class="social-media-list">
          <!--<li>Email: seonghwan_jun AT urmc DOT rochester DOT edu</li>-->

          
          <li>
            <a href="https://github.com/junseonghwan">
              <span class="icon  icon--github">
                <svg viewBox="0 0 16 16">
                  <path fill="#828282" d="M7.999,0.431c-4.285,0-7.76,3.474-7.76,7.761 c0,3.428,2.223,6.337,5.307,7.363c0.388,0.071,0.53-0.168,0.53-0.374c0-0.184-0.007-0.672-0.01-1.32 c-2.159,0.469-2.614-1.04-2.614-1.04c-0.353-0.896-0.862-1.135-0.862-1.135c-0.705-0.481,0.053-0.472,0.053-0.472 c0.779,0.055,1.189,0.8,1.189,0.8c0.692,1.186,1.816,0.843,2.258,0.645c0.071-0.502,0.271-0.843,0.493-1.037 C4.86,11.425,3.049,10.76,3.049,7.786c0-0.847,0.302-1.54,0.799-2.082C3.768,5.507,3.501,4.718,3.924,3.65 c0,0,0.652-0.209,2.134,0.796C6.677,4.273,7.34,4.187,8,4.184c0.659,0.003,1.323,0.089,1.943,0.261 c1.482-1.004,2.132-0.796,2.132-0.796c0.423,1.068,0.157,1.857,0.077,2.054c0.497,0.542,0.798,1.235,0.798,2.082 c0,2.981-1.814,3.637-3.543,3.829c0.279,0.24,0.527,0.713,0.527,1.437c0,1.037-0.01,1.874-0.01,2.129 c0,0.208,0.14,0.449,0.534,0.373c3.081-1.028,5.302-3.935,5.302-7.362C15.76,3.906,12.285,0.431,7.999,0.431z"/>
                </svg>
              </span>

              <span class="username">junseonghwan</span>
            </a>
          </li>
          

          
        </ul>
      </div>
<!--
      <div class="footer-col  footer-col-3">
        <p class="text"></p>
      </div>
    </div>
-->
  </div>

</footer>


  </body>

</html>
