<!DOCTYPE html>
<html>

  <head>
  <meta charset="utf-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="viewport" content="width=device-width, initial-scale=1">

  <title>Configuring TitanCNA (manually)</title>
  <meta name="description" content="TitanCNA ships with various workflows that seem rather easy to configure and run. But as a newbie in bioinformatics, I consider it to be a valuable exercise ...">

  <link rel="stylesheet" href="/css/main.css">
  <link rel="stylesheet" href="/css/bootstrap.css">
  <link rel="canonical" href="junseonghwan.github.io/2018/10/08/titancna.html">
  <link rel="alternate" type="application/rss+xml" title="Seong-Hwan Jun" href="junseonghwan.github.io/feed.xml" />

  

  <script type="text/javascript" async
    src="https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-MML-AM_CHTML">
  </script>
  <!--
  <script src="https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML" type="text/javascript"></script>-->

</head>


  <body>

    <header class="site-header">

  <div class="wrapper">

    <a class="site-title" href="/">Seong-Hwan Jun</a>

    <nav class="site-nav">
      <a href="#" class="menu-icon">
        <svg viewBox="0 0 18 15">
          <path fill="#424242" d="M18,1.484c0,0.82-0.665,1.484-1.484,1.484H1.484C0.665,2.969,0,2.304,0,1.484l0,0C0,0.665,0.665,0,1.484,0 h15.031C17.335,0,18,0.665,18,1.484L18,1.484z"/>
          <path fill="#424242" d="M18,7.516C18,8.335,17.335,9,16.516,9H1.484C0.665,9,0,8.335,0,7.516l0,0c0-0.82,0.665-1.484,1.484-1.484 h15.031C17.335,6.031,18,6.696,18,7.516L18,7.516z"/>
          <path fill="#424242" d="M18,13.516C18,14.335,17.335,15,16.516,15H1.484C0.665,15,0,14.335,0,13.516l0,0 c0-0.82,0.665-1.484,1.484-1.484h15.031C17.335,12.031,18,12.696,18,13.516L18,13.516z"/>
        </svg>
      </a>

      <div class="trigger">
        
          
          <a class="page-link" href="/code/">Code</a>
          
        
          
        
          
        
          
          <a class="page-link" href="/research/">Research</a>
          
        
      </div>
    </nav>

  </div>

</header>


    <div class="page-content">
      <div class="wrapper">
        <div class="post">

  <header class="post-header">
    <h1 class="post-title">Configuring TitanCNA (manually)</h1>
    <p class="post-meta">Oct 8, 2018</p>
  </header>

  <article class="post-content">
    <p>TitanCNA ships with various workflows that seem rather easy to configure and run. But as a newbie in bioinformatics, I consider it to be a valuable exercise to configure and get tools working from ground up. This post assumes the following starting point:</p>
<ol>
  <li>BAM files for Normal and Tumor samples that have been through the standard pre-processing pipeline, i.e., mapping, sorting, marking duplicate, adding read groups, and indel realignment.</li>
  <li>Germline SNP calling using <a href="https://software.broadinstitute.org/gatk/documentation/tooldocs/3.8-0/org_broadinstitute_gatk_tools_walkers_haplotypecaller_HaplotypeCaller.php"><code class="language-plaintext highlighter-rouge">HaplotypeCaller</code></a>.</li>
  <li>Softwares:
    <ul>
      <li><a href="https://github.com/cancerit/alleleCount">alleleCount</a>.</li>
      <li><a href="https://github.com/shahcompbio/hmmcopy_utils">HMMcopy_utils</a>, and</li>
      <li><a href="http://bioconductor.org/packages/release/bioc/html/HMMcopy.html">HMMcopy</a>,</li>
      <li><a href="https://bioconductor.org/packages/release/bioc/html/TitanCNA.html">TitanCNA</a>,</li>
    </ul>
  </li>
</ol>

<h2 id="analysis-workflow">Analysis workflow</h2>
<ol>
  <li>Identify germline SNP positions \(L = \{ t_i\}_{i=1}^{T}\), where \(T\) denotes the total number of germline SNP positions. We want to extract loci.
    <ul>
      <li>Input: BAM file for normal sample.</li>
      <li>Output:
        <ol>
          <li>Loci file, which is a tab separated file with two columns, <code class="language-plaintext highlighter-rouge">chr</code>, <code class="language-plaintext highlighter-rouge">posn</code>. The file is not to contain any header.</li>
          <li>TSV file with four columns, <code class="language-plaintext highlighter-rouge">chr</code>, <code class="language-plaintext highlighter-rouge">posn</code>, <code class="language-plaintext highlighter-rouge">ref</code>, <code class="language-plaintext highlighter-rouge">nonRef</code>.</li>
        </ol>
      </li>
    </ul>
  </li>
  <li>Get allelic counts from normal and tumor samples. We will use <code class="language-plaintext highlighter-rouge">alleleCount</code> program, which accepts tab separated file <code class="language-plaintext highlighter-rouge">chr</code> and <code class="language-plaintext highlighter-rouge">posn</code> (without column header).
    <ul>
      <li>Input: Output of Step 1.</li>
      <li>Output: Data file containing six columns, <code class="language-plaintext highlighter-rouge">chr</code>, <code class="language-plaintext highlighter-rouge">posn</code>, <code class="language-plaintext highlighter-rouge">ref</code>, <code class="language-plaintext highlighter-rouge">refCount</code>, <code class="language-plaintext highlighter-rouge">nonRef</code>, <code class="language-plaintext highlighter-rouge">nonRefCount</code>. Output name: <code class="language-plaintext highlighter-rouge">Normal/allele_counts.txt</code>, <code class="language-plaintext highlighter-rouge">Tumor/allele_counts.txt</code>.</li>
    </ul>
  </li>
  <li>Correction for GC content and mappability biases. The raw counts and depths may be biased due to GC content. This is where <code class="language-plaintext highlighter-rouge">HMMcopy_util</code> will be useful.
    <ul>
      <li>Input: BAM file for Normal and Tumor samples</li>
      <li>Output: WIG file for read counts computed by binning Normal and Tumor samples at 1000kb. Output names: <code class="language-plaintext highlighter-rouge">Normal/readcounts.seg</code>, <code class="language-plaintext highlighter-rouge">Tumor/readcounts.seg</code>.</li>
    </ul>
  </li>
  <li>Perform CNA.
    <ul>
      <li>Input: <code class="language-plaintext highlighter-rouge">Tumor/allele_counts.txt</code>, <code class="language-plaintext highlighter-rouge">Normal/readcounts.seg</code>, <code class="language-plaintext highlighter-rouge">Tumor/readcounts.seg</code>.</li>
      <li>Output: Data file containing copy number profile intervals.</li>
    </ul>
  </li>
  <li>Repeat CNA for a range of clonal clusters.</li>
</ol>

<h3 id="step-1-identify-germline-snp-positions-from-normal-sample">Step 1: Identify germline SNP positions from normal sample</h3>
<ul>
  <li><strong>Input</strong>: BAM file for normal sample.</li>
  <li><strong>Output</strong>: To produce file with two columns, <code class="language-plaintext highlighter-rouge">chr</code>, <code class="language-plaintext highlighter-rouge">posn</code> and another file with four columns <code class="language-plaintext highlighter-rouge">chr</code>, <code class="language-plaintext highlighter-rouge">posn</code>, <code class="language-plaintext highlighter-rouge">ref</code>, <code class="language-plaintext highlighter-rouge">alt</code>.</li>
</ul>

<p>We essentially need to write a simple Python script to process the output of <code class="language-plaintext highlighter-rouge">HaplotypeCaller</code>. Essentially, reading each line and retrieving the <code class="language-plaintext highlighter-rouge">chr</code> and <code class="language-plaintext highlighter-rouge">posn</code> of germline SNP. Python code for this procedure is avilable on <a href="https://github.com/junseonghwan/scripts">my Github</a>.</p>

<h4 id="notes">Notes</h4>
<ul>
  <li>Q: are the germline SNP loci specific to the sample or should we use a set of known positions?
    <ul>
      <li>A1: TitanCNA vignette reads: “The data points of interest are the germline heterozygous SNP loci identified in the matched normal WGS sample. This is a pre-processing step that is completed by the user, and is outside of the scope of this R package.”</li>
      <li>A2: One can use a known loci if desired. But in the dataset that I am interested in analyzing, what I have is exome data. Therefore, some of these known loci may not have any read (or only a few), which would definitely lead to misleading copy number inference.</li>
    </ul>
  </li>
</ul>

<h3 id="step-2-extract-read-depth-from-tumorand-normal-samples">Step 2: Extract read depth from tumor and normal samples</h3>
<ul>
  <li><strong>Input</strong>: loci file from Step 1.</li>
  <li><strong>Output</strong>: A file containing six columns: <code class="language-plaintext highlighter-rouge">chr, posn, ref, refOriginal, nonRef, tumDepth</code>, for normal and tumor sample.</li>
</ul>

<p>We will use <code class="language-plaintext highlighter-rouge">alleleCounter</code> to obtain the count at each of the locus. This is just a one line script (assuming alleleCounter is installed),</p>
<div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code>alleleCounter <span class="nt">-l</span> path/to/loci_file_from_step1 <span class="nt">-r</span> path/to/hg/ <span class="nt">-b</span> <span class="nv">$1</span>/indel_realigned.bam <span class="nt">-o</span> <span class="nv">$1</span>/sample.allelecount
</code></pre></div></div>
<p>where <code class="language-plaintext highlighter-rouge">$1</code> indicates the directory containing the BAM file and where the output is to be stored.</p>

<p>The output of <code class="language-plaintext highlighter-rouge">alleleCounter</code> needs to be further processed to form the desired output containing 6 columns. To do so, we will use the second output from Step 1 (i.e., the larger file containing 4 columns). I have written a simple script that does this. It requires <code class="language-plaintext highlighter-rouge">helper_functions.py</code>, which is available <a href="https://github.com/junseonghwan/scripts/blob/master/pipeline4/helper_functions.py">here</a>.</p>

<div class="language-python highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="kn">import</span> <span class="n">argparse</span>
<span class="kn">import</span> <span class="n">os</span>

<span class="kn">import</span> <span class="n">pandas</span> <span class="k">as</span> <span class="n">pd</span>

<span class="kn">import</span> <span class="n">helper_functions</span> <span class="k">as</span> <span class="n">helpers</span>

<span class="n">parser</span> <span class="o">=</span> <span class="n">argparse</span><span class="p">.</span><span class="nc">ArgumentParser</span><span class="p">(</span><span class="n">description</span><span class="o">=</span><span class="s">'Generate input file for TitanCNA.'</span><span class="p">)</span>
<span class="n">parser</span><span class="p">.</span><span class="nf">add_argument</span><span class="p">(</span><span class="s">'study_dir'</span><span class="p">,</span> <span class="n">help</span><span class="o">=</span><span class="s">"Name of the directory containing the data files for a study. E.g., TNBC"</span><span class="p">,</span> <span class="nb">type</span><span class="o">=</span><span class="nb">str</span><span class="p">,</span> <span class="n">default</span><span class="o">=</span><span class="s">"TNBC"</span><span class="p">)</span>
<span class="n">parser</span><span class="p">.</span><span class="nf">add_argument</span><span class="p">(</span><span class="s">'patient_id'</span><span class="p">,</span> <span class="n">help</span><span class="o">=</span><span class="s">"Patient ID. E.g., KTN126"</span><span class="p">,</span> <span class="nb">type</span><span class="o">=</span><span class="nb">str</span><span class="p">)</span>
<span class="n">parser</span><span class="p">.</span><span class="nf">add_argument</span><span class="p">(</span><span class="s">'--data_dir'</span><span class="p">,</span> <span class="n">help</span><span class="o">=</span><span class="s">"Data directory."</span><span class="p">,</span> <span class="nb">type</span><span class="o">=</span><span class="nb">str</span><span class="p">,</span> <span class="n">default</span><span class="o">=</span><span class="s">"/proj/uppstore2018155/"</span><span class="p">)</span>
<span class="n">parser</span><span class="p">.</span><span class="nf">add_argument</span><span class="p">(</span><span class="s">'--sample_name'</span><span class="p">,</span> <span class="n">help</span><span class="o">=</span><span class="s">"Name of the directory containing the sample to be processed. E.g., KTN1260."</span><span class="p">,</span> <span class="nb">type</span><span class="o">=</span><span class="nb">str</span><span class="p">,</span> <span class="n">default</span><span class="o">=</span><span class="s">""</span><span class="p">)</span>
<span class="n">parser</span><span class="p">.</span><span class="nf">add_argument</span><span class="p">(</span><span class="s">'--germline_snv_file'</span><span class="p">,</span> <span class="n">help</span><span class="o">=</span><span class="s">"Name of the germline SNV file."</span><span class="p">,</span> <span class="nb">type</span><span class="o">=</span><span class="nb">str</span><span class="p">,</span> <span class="n">default</span><span class="o">=</span><span class="s">"germline_snvs.tsv"</span><span class="p">)</span>
<span class="n">args</span> <span class="o">=</span> <span class="n">parser</span><span class="p">.</span><span class="nf">parse_args</span><span class="p">()</span>

<span class="n">patient_dir</span> <span class="o">=</span> <span class="n">os</span><span class="p">.</span><span class="n">path</span><span class="p">.</span><span class="nf">join</span><span class="p">(</span><span class="n">args</span><span class="p">.</span><span class="n">data_dir</span><span class="p">,</span> <span class="n">args</span><span class="p">.</span><span class="n">study_dir</span><span class="p">,</span> <span class="n">args</span><span class="p">.</span><span class="n">patient_id</span><span class="p">)</span>
<span class="n">common_dir</span> <span class="o">=</span> <span class="n">os</span><span class="p">.</span><span class="n">path</span><span class="p">.</span><span class="nf">join</span><span class="p">(</span><span class="n">patient_dir</span><span class="p">,</span> <span class="s">"common"</span><span class="p">)</span>
<span class="n">file_handle</span> <span class="o">=</span> <span class="nf">open</span><span class="p">(</span><span class="n">os</span><span class="p">.</span><span class="n">path</span><span class="p">.</span><span class="nf">join</span><span class="p">(</span><span class="n">common_dir</span><span class="p">,</span> <span class="n">args</span><span class="p">.</span><span class="n">germline_snv_file</span><span class="p">),</span> <span class="s">'r'</span><span class="p">)</span>
<span class="nf">next</span><span class="p">(</span><span class="n">file_handle</span><span class="p">)</span>
<span class="n">loci_dict</span> <span class="o">=</span> <span class="nf">dict</span><span class="p">()</span>
<span class="k">for</span> <span class="n">line</span> <span class="ow">in</span> <span class="n">file_handle</span><span class="p">:</span>
    <span class="n">row</span> <span class="o">=</span> <span class="n">line</span><span class="p">.</span><span class="nf">split</span><span class="p">(</span><span class="s">'</span><span class="se">\t</span><span class="s">'</span><span class="p">)</span>
    <span class="n">key</span> <span class="o">=</span> <span class="nf">str</span><span class="p">(</span><span class="n">row</span><span class="p">[</span><span class="mi">0</span><span class="p">])</span> <span class="o">+</span> <span class="s">"_"</span> <span class="o">+</span> <span class="nf">str</span><span class="p">(</span><span class="n">row</span><span class="p">[</span><span class="mi">1</span><span class="p">])</span>
    <span class="n">loci_dict</span><span class="p">[</span><span class="n">key</span><span class="p">]</span> <span class="o">=</span> <span class="p">(</span><span class="n">row</span><span class="p">[</span><span class="mi">2</span><span class="p">].</span><span class="nf">strip</span><span class="p">(),</span> <span class="n">row</span><span class="p">[</span><span class="mi">3</span><span class="p">].</span><span class="nf">strip</span><span class="p">())</span>

<span class="k">if</span> <span class="n">args</span><span class="p">.</span><span class="n">sample_name</span> <span class="o">==</span> <span class="s">""</span><span class="p">:</span>
    <span class="n">samples</span> <span class="o">=</span> <span class="n">os</span><span class="p">.</span><span class="nf">listdir</span><span class="p">(</span><span class="n">patient_dir</span><span class="p">)</span>
<span class="k">else</span><span class="p">:</span>
    <span class="n">samples</span> <span class="o">=</span> <span class="p">[</span><span class="n">args</span><span class="p">.</span><span class="n">sample_name</span><span class="p">]</span>

<span class="k">for</span> <span class="n">sample</span> <span class="ow">in</span> <span class="n">samples</span><span class="p">:</span>
    <span class="k">if</span> <span class="n">sample</span><span class="p">.</span><span class="nf">startswith</span><span class="p">(</span><span class="s">"."</span><span class="p">)</span> <span class="ow">or</span> <span class="n">sample</span> <span class="o">==</span> <span class="s">"common"</span><span class="p">:</span>
        <span class="k">continue</span>
    <span class="n">sample_dir</span> <span class="o">=</span> <span class="n">os</span><span class="p">.</span><span class="n">path</span><span class="p">.</span><span class="nf">join</span><span class="p">(</span><span class="n">patient_dir</span><span class="p">,</span> <span class="n">sample</span><span class="p">)</span>
    <span class="n">count_file</span> <span class="o">=</span> <span class="n">os</span><span class="p">.</span><span class="n">path</span><span class="p">.</span><span class="nf">join</span><span class="p">(</span><span class="n">sample_dir</span><span class="p">,</span> <span class="s">"sample.allelecount"</span><span class="p">)</span>
    <span class="n">file_handle</span> <span class="o">=</span> <span class="nf">open</span><span class="p">(</span><span class="n">count_file</span><span class="p">,</span> <span class="s">'r'</span><span class="p">)</span>
    <span class="n">dat</span> <span class="o">=</span> <span class="nf">list</span><span class="p">()</span>
    <span class="nf">next</span><span class="p">(</span><span class="n">file_handle</span><span class="p">)</span>
    <span class="k">for</span> <span class="n">line</span> <span class="ow">in</span> <span class="n">file_handle</span><span class="p">:</span>
        <span class="n">row</span> <span class="o">=</span> <span class="n">line</span><span class="p">.</span><span class="nf">split</span><span class="p">(</span><span class="s">'</span><span class="se">\t</span><span class="s">'</span><span class="p">)</span>
        <span class="n">key</span> <span class="o">=</span> <span class="nf">str</span><span class="p">(</span><span class="n">row</span><span class="p">[</span><span class="mi">0</span><span class="p">])</span> <span class="o">+</span> <span class="s">"_"</span> <span class="o">+</span> <span class="nf">str</span><span class="p">(</span><span class="n">row</span><span class="p">[</span><span class="mi">1</span><span class="p">])</span>
        <span class="k">if</span> <span class="n">key</span> <span class="ow">in</span> <span class="n">loci_dict</span><span class="p">:</span>
            <span class="n">ref</span><span class="p">,</span> <span class="n">alt</span> <span class="o">=</span> <span class="n">loci_dict</span><span class="p">[</span><span class="n">key</span><span class="p">]</span>
            <span class="k">if</span> <span class="nf">len</span><span class="p">(</span><span class="n">ref</span><span class="p">.</span><span class="nf">strip</span><span class="p">())</span> <span class="o">&gt;</span> <span class="mi">1</span> <span class="ow">or</span> <span class="nf">len</span><span class="p">(</span><span class="n">alt</span><span class="p">.</span><span class="nf">strip</span><span class="p">())</span> <span class="o">&gt;</span> <span class="mi">1</span><span class="p">:</span>
                <span class="k">continue</span>
            <span class="n">ref_count</span> <span class="o">=</span> <span class="nf">int</span><span class="p">(</span><span class="n">row</span><span class="p">[</span><span class="mi">2</span> <span class="o">+</span> <span class="n">helpers</span><span class="p">.</span><span class="nf">get_idx</span><span class="p">(</span><span class="n">ref</span><span class="p">)])</span>
            <span class="n">alt_count</span> <span class="o">=</span> <span class="nf">int</span><span class="p">(</span><span class="n">row</span><span class="p">[</span><span class="mi">2</span> <span class="o">+</span> <span class="n">helpers</span><span class="p">.</span><span class="nf">get_idx</span><span class="p">(</span><span class="n">alt</span><span class="p">)])</span>
            <span class="n">dat</span><span class="p">.</span><span class="nf">append</span><span class="p">((</span><span class="nf">str</span><span class="p">(</span><span class="n">row</span><span class="p">[</span><span class="mi">0</span><span class="p">]),</span> <span class="nf">int</span><span class="p">(</span><span class="n">row</span><span class="p">[</span><span class="mi">1</span><span class="p">]),</span> <span class="n">ref</span><span class="p">,</span> <span class="n">ref_count</span><span class="p">,</span> <span class="n">alt</span><span class="p">,</span> <span class="n">ref_count</span> <span class="o">+</span> <span class="n">alt_count</span><span class="p">))</span>

    <span class="n">df</span> <span class="o">=</span> <span class="n">pd</span><span class="p">.</span><span class="nc">DataFrame</span><span class="p">(</span><span class="n">data</span><span class="o">=</span><span class="n">dat</span><span class="p">,</span> <span class="n">columns</span><span class="o">=</span><span class="p">[</span><span class="s">'chr'</span><span class="p">,</span> <span class="s">'posn'</span><span class="p">,</span> <span class="s">'ref'</span><span class="p">,</span> <span class="s">'refOriginal'</span><span class="p">,</span> <span class="s">'nonRef'</span><span class="p">,</span> <span class="s">'tumDepth'</span><span class="p">])</span>
    <span class="c1"># save to file
</span>    <span class="n">dest_file_name</span> <span class="o">=</span> <span class="n">os</span><span class="p">.</span><span class="n">path</span><span class="p">.</span><span class="nf">join</span><span class="p">(</span><span class="n">sample_dir</span><span class="p">,</span> <span class="s">"allele_counts.txt"</span><span class="p">)</span>
    <span class="n">df</span><span class="p">.</span><span class="nf">to_csv</span><span class="p">(</span><span class="n">dest_file_name</span><span class="p">,</span> <span class="n">sep</span><span class="o">=</span><span class="s">"</span><span class="se">\t</span><span class="s">"</span><span class="p">,</span> <span class="n">index</span><span class="o">=</span><span class="bp">False</span><span class="p">)</span>
</code></pre></div></div>

<h4 id="notes-1">Notes</h4>
<ul>
  <li>Out of curiosity, I wanted to compare the output of <code class="language-plaintext highlighter-rouge">HaplotypeCaller</code> to the counts retrieved by <a href="https://pysam.readthedocs.io/en/latest/api.html">pysam</a> and <code class="language-plaintext highlighter-rouge">alleleCount</code>.
    <ul>
      <li>Note that <code class="language-plaintext highlighter-rouge">HaplotypeCaller</code> performs indel realignment internally. To ensure there is as little discrepancy as possible, we should use the BAM file with indel realignment.</li>
      <li>I chose <code class="language-plaintext highlighter-rouge">pysam</code> because it seemed easy to set up and use and I have been told that it is pretty fast.</li>
    </ul>
  </li>
</ul>

<p>To retrieve the reads at a specific loci, one can use following code:</p>
<div class="language-python highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="kn">import</span> <span class="n">pysam</span>
<span class="n">bamfile</span> <span class="o">=</span> <span class="n">pysam</span><span class="p">.</span><span class="nc">AlignmentFile</span><span class="p">(</span><span class="s">"path/to/bam"</span><span class="p">)</span>
<span class="c1"># Example: to get the reads at 1:949597
</span><span class="n">chromosome</span> <span class="o">=</span> <span class="s">'1'</span>
<span class="n">start</span> <span class="o">=</span> <span class="mi">949597</span>
<span class="n">ref</span> <span class="o">=</span> <span class="s">'C'</span>
<span class="n">alt</span> <span class="o">=</span> <span class="s">'T'</span>
<span class="n">bamfile</span><span class="p">.</span><span class="nf">count</span><span class="p">(</span><span class="s">'1'</span><span class="p">,</span> <span class="n">start</span><span class="o">-</span><span class="mi">1</span><span class="p">,</span> <span class="n">start</span><span class="p">)</span>
<span class="n">ref_count</span> <span class="o">=</span> <span class="mi">0</span>
<span class="n">alt_count</span> <span class="o">=</span> <span class="mi">0</span>
<span class="n">other_count</span> <span class="o">=</span> <span class="mi">0</span>
<span class="k">for</span> <span class="n">read</span> <span class="ow">in</span> <span class="n">bamfile</span><span class="p">.</span><span class="nf">fetch</span><span class="p">(</span><span class="s">'1'</span><span class="p">,</span> <span class="n">start</span><span class="o">-</span><span class="mi">1</span><span class="p">,</span> <span class="n">start</span><span class="p">):</span>
    <span class="n">seq</span> <span class="o">=</span> <span class="n">read</span><span class="p">.</span><span class="nf">get_forward_sequence</span><span class="p">()</span>
    <span class="n">quals</span> <span class="o">=</span> <span class="n">read</span><span class="p">.</span><span class="nf">get_forward_qualities</span><span class="p">()</span>
    <span class="n">pos</span> <span class="o">=</span> <span class="n">start</span> <span class="o">-</span> <span class="mi">1</span> <span class="o">-</span> <span class="n">read</span><span class="p">.</span><span class="n">reference_start</span>
    <span class="k">if</span> <span class="n">quals</span><span class="p">[</span><span class="n">pos</span><span class="p">]</span> <span class="o">&gt;</span> <span class="mi">40</span><span class="p">:</span>
        <span class="k">if</span> <span class="n">seq</span><span class="p">[</span><span class="n">pos</span><span class="p">]</span> <span class="o">==</span> <span class="n">ref</span><span class="p">:</span>
            <span class="n">ref_count</span> <span class="o">=</span> <span class="n">ref_count</span> <span class="o">+</span> <span class="mi">1</span>
            <span class="c1">#print("%s, %i" % (seq[pos], quals[pos]))
</span>        <span class="k">elif</span> <span class="n">seq</span><span class="p">[</span><span class="n">pos</span><span class="p">]</span> <span class="o">==</span> <span class="n">alt</span><span class="p">:</span>
            <span class="n">alt_count</span> <span class="o">=</span> <span class="n">alt_count</span> <span class="o">+</span> <span class="mi">1</span>
            <span class="c1">#print("%s, %i" % (seq[pos], quals[pos]))
</span>        <span class="k">else</span><span class="p">:</span>
            <span class="n">other_count</span> <span class="o">=</span> <span class="n">other_count</span> <span class="o">+</span> <span class="mi">1</span>
<span class="nf">print</span><span class="p">(</span><span class="s">"%i, %i, %i"</span> <span class="o">%</span> <span class="p">(</span><span class="n">ref_count</span><span class="p">,</span> <span class="n">alt_count</span><span class="p">,</span> <span class="n">other_count</span><span class="p">))</span>
<span class="c1"># Output:
# 33, 15, 10
</span></code></pre></div></div>
<p>Running the above code produced 33, 15, 10 as the output for the reference, alternate, and other bases. On the contrary, VCF file produced by <code class="language-plaintext highlighter-rouge">HaplotypeCaller</code> for chromosome 1 had 16 and 13 as counts for reference and alternate alleles. The discrepancy in the counts is too large to be ignored, prompting <a href="https://gatkforums.broadinstitute.org/gatk/discussion/6694/determining-read-depth-from-bam-file-4-methods-3-different-results">further investigation</a>.</p>
<ul>
  <li>In short, it can be due to the duplicates, which are marked but may not have been removed by GATK/Picard tool. <code class="language-plaintext highlighter-rouge">HaplotypeCaller</code> knows to ignore them, but <code class="language-plaintext highlighter-rouge">pysam</code> may not.</li>
  <li>Not satisfied, I used <code class="language-plaintext highlighter-rouge">alleleCounter</code> to compute the counts at loci identified by <code class="language-plaintext highlighter-rouge">HaplotypeCaller</code>. It gave counts that were very close to HaplotypeCaller. For example, at position 949597, I got 16 and 12 as counts for reference and alternate allele compared to 16 and 13 from HaplotypeCaller. In general, the counts were more or less similar at different locations as well.</li>
  <li><strong>Conclusion:</strong> I decided to go with <code class="language-plaintext highlighter-rouge">alleleCounter</code>.
    <ul>
      <li>Disclaimer: I’m not advocating for or against any tool. I’m sure there must be a way to configure <code class="language-plaintext highlighter-rouge">pysam</code> to behave in the desired way. It’s just that I didn’t investigate and I was satisfied with <code class="language-plaintext highlighter-rouge">alleleCounter</code>.</li>
    </ul>
  </li>
</ul>

<h3 id="step-3-correction-for-gc-content-and-mappability-biases">Step 3: Correction for GC content and mappability biases</h3>
<p><code class="language-plaintext highlighter-rouge">HMMcopy</code> is a package developed for correcting for GC content and mappability biases. As per TitanCNA vignette, TitanCNA seems to work extensively with WIG file formats:</p>

<p>“For TITAN, the user will need to generate the read depth files for the tumour and normal samples in the form of WIG files.”</p>

<p>We will use <code class="language-plaintext highlighter-rouge">HMMcopy_util</code> to obtain the read depth files for tumour and normal samples in WIG format.</p>

<h3 id="step-3-1-hmmcopy_util">Step 3-1: <code class="language-plaintext highlighter-rouge">HMMcopy_util</code></h3>
<p>Clone <code class="language-plaintext highlighter-rouge">HMMcopy_util</code> via command <code class="language-plaintext highlighter-rouge">git clone https://github.com/shahcompbio/hmmcopy_utils</code>.  Follow <a href="https://github.com/shahcompbio/hmmcopy_utils">instructions</a> to compile the program. This requires <code class="language-plaintext highlighter-rouge">cmake</code>, which you can get using <code class="language-plaintext highlighter-rouge">brew</code> on OSX and <code class="language-plaintext highlighter-rouge">make</code>, which should be available as long as you have installed Xcode. There are 3 programs in <code class="language-plaintext highlighter-rouge">HMMcopy_utils</code>:</p>
<ul>
  <li><code class="language-plaintext highlighter-rouge">bin/readCounter</code> produces WIG file from your (tumor and normal) BAM file. Use the window size of 1,000 (this is the recommended option). I added the <code class="language-plaintext highlighter-rouge">hmmcopy_utils/bin/</code> to <code class="language-plaintext highlighter-rouge">PATH</code> in <code class="language-plaintext highlighter-rouge">.bashrc</code>. With that, I can issue the following command,
    <div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code>readCounter <span class="nt">-w</span> 1000 path/to/normal/indel_realigned.bam <span class="o">&gt;</span> path/to/normal/readcounts.seg
readCounter <span class="nt">-w</span> 1000 path/to/tumor/indel_realigned.bam <span class="o">&gt;</span> path/to/tumor/readcounts.seg
</code></pre></div>    </div>
  </li>
  <li><code class="language-plaintext highlighter-rouge">bin/gcCounter</code> produces WIG file from reference human genome <code class="language-plaintext highlighter-rouge">fasta</code> file. The HG file should be the same one that we have been using in the pre-processing steps.</li>
  <li><code class="language-plaintext highlighter-rouge">bin/mapCounter</code> produces WIG file. It accepts BigWig file format for the reference hg file.</li>
</ul>

<h4 id="notes-2">Notes</h4>
<ul>
  <li><code class="language-plaintext highlighter-rouge">HMMcopy_util</code> seems to ship GC and Mappability WIG files generated with window size of 1,000. Therefore,<code class="language-plaintext highlighter-rouge">bin/gcCounter</code> and <code class="language-plaintext highlighter-rouge">bin/mapCounter</code> need not be run unless one wishes to use different window size.</li>
  <li>Question: how does one obtain BigWig file from HG18/19?
    <ul>
      <li>Some hint <a href="https://github.com/broadinstitute/ichorCNA/issues/29">here</a>?</li>
    </ul>
  </li>
</ul>

<h3 id="step-3-2-hmmcopy">Step 3-2: <code class="language-plaintext highlighter-rouge">HMMcopy</code></h3>
<p>This step can be skipped since TitanCNA has a wrapper function that calls the functions in <code class="language-plaintext highlighter-rouge">HMMcopy</code> but it is nice to try out the package to see exactly what it does and to catch any problems that may have occurred up to this point in the processing steps. They have a nice <a href="https://www.bioconductor.org/packages/devel/bioc/vignettes/HMMcopy/inst/doc/HMMcopy.pdf">vignette</a> that is easy to follow, which I paste below:</p>
<div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">library</span><span class="p">(</span><span class="n">HMMcopy</span><span class="p">)</span><span class="w">
</span><span class="n">readfile</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="s2">"path/to/output/of/readCounter"</span><span class="w">
</span><span class="n">gcfile</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="s2">"path/to/hmmcopy_utils/data/gc_hg19.wig"</span><span class="w">
</span><span class="n">mapfile</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="s2">"path/to/hmmcopy_utils/data/map_hg19.wig"</span><span class="w">
</span><span class="n">uncorrected_reads</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">wigsToRangedData</span><span class="p">(</span><span class="n">readfile</span><span class="p">,</span><span class="w"> </span><span class="n">gcfile</span><span class="p">,</span><span class="w"> </span><span class="n">mapfile</span><span class="p">)</span><span class="w">
</span><span class="n">corrected_copy</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">correctReadcount</span><span class="p">(</span><span class="n">uncorrected_reads</span><span class="p">)</span><span class="w">
</span><span class="n">segmented_copy</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">HMMsegment</span><span class="p">(</span><span class="n">corrected_copy</span><span class="p">)</span><span class="w">

</span><span class="n">par</span><span class="p">(</span><span class="n">cex.main</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">0.7</span><span class="p">,</span><span class="w"> </span><span class="n">cex.lab</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">0.7</span><span class="p">,</span><span class="w"> </span><span class="n">cex.axis</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">0.7</span><span class="p">,</span><span class="w"> </span><span class="n">mar</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="m">4</span><span class="p">,</span><span class="w"> </span><span class="m">4</span><span class="p">,</span><span class="w"> </span><span class="m">2</span><span class="p">,</span><span class="w"> </span><span class="m">0.5</span><span class="p">))</span><span class="w">
</span><span class="n">plotBias</span><span class="p">(</span><span class="n">corrected_copy</span><span class="p">,</span><span class="w"> </span><span class="n">pch</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">20</span><span class="p">,</span><span class="w"> </span><span class="n">cex</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">0.5</span><span class="p">)</span><span class="w">

</span><span class="n">par</span><span class="p">(</span><span class="n">mar</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="m">4</span><span class="p">,</span><span class="w"> </span><span class="m">4</span><span class="p">,</span><span class="w"> </span><span class="m">2</span><span class="p">,</span><span class="w"> </span><span class="m">0</span><span class="p">))</span><span class="w">
</span><span class="n">plotCorrection</span><span class="p">(</span><span class="n">corrected_copy</span><span class="p">,</span><span class="w"> </span><span class="n">pch</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"."</span><span class="p">)</span><span class="w">

</span><span class="n">par</span><span class="p">(</span><span class="n">cex.main</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">0.7</span><span class="p">,</span><span class="w"> </span><span class="n">cex.lab</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">0.7</span><span class="p">,</span><span class="w"> </span><span class="n">cex.axis</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">0.7</span><span class="p">,</span><span class="w"> </span><span class="n">mar</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="m">4</span><span class="p">,</span><span class="w"> </span><span class="m">4</span><span class="p">,</span><span class="w"> </span><span class="m">2</span><span class="p">,</span><span class="w"> </span><span class="m">0.5</span><span class="p">))</span><span class="w">
</span><span class="n">plotSegments</span><span class="p">(</span><span class="n">corrected_copy</span><span class="p">,</span><span class="w"> </span><span class="n">segmented_copy</span><span class="p">)</span><span class="w">
</span></code></pre></div></div>

<h4 id="notes-3">Notes</h4>
<ul>
  <li>For TNBC data (exome data), I was not able to generate a nice figure as is given in the <code class="language-plaintext highlighter-rouge">HMMcopy</code> vignette. This is because there are many regions with little or no reads and the scale for the figure is exorted as a result.</li>
</ul>

<p><img src="/images/hmmcopy_Rplot.png" alt="HMMcopy Bias plot" /></p>

<h3 id="step-4-titancna">Step 4: TitanCNA</h3>
<p>** Input: <code class="language-plaintext highlighter-rouge">Normal/allele_counts.txt</code> and <code class="language-plaintext highlighter-rouge">Tumor/allele_counts.txt</code> from Step 2.
** Output: Data file of copy number profile.</p>

<p>The steps for using TitanCNA are described in Section 5 of the vignette [2]. I have adapted this code for analysis of TNBC data. The complete code is available <a href="/resources/titan_workflow.txt">here</a>. Here, I will briefly describe the steps that I have taken to prepare the input data. I chose not to perform GC correction for the time being.</p>

<div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">library</span><span class="p">(</span><span class="n">TitanCNA</span><span class="p">)</span><span class="w">
</span><span class="n">library</span><span class="p">(</span><span class="n">HMMcopy</span><span class="p">)</span><span class="w">

</span><span class="n">min_depth</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="m">10</span><span class="w">
</span><span class="n">max_depth</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="m">500</span><span class="w">

</span><span class="n">infile</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="s2">"path/to/Normal/allele_counts.txt"</span><span class="w">
</span><span class="n">normal_data</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">read.csv</span><span class="p">(</span><span class="n">infile</span><span class="p">,</span><span class="w"> </span><span class="n">header</span><span class="o">=</span><span class="nb">T</span><span class="p">,</span><span class="w"> </span><span class="n">sep</span><span class="o">=</span><span class="s2">"\t"</span><span class="p">)</span><span class="w">

</span><span class="n">infile</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="s2">"path/to/Tumor/allele_counts.txt"</span><span class="w">
</span><span class="n">tumor_data</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">read.csv</span><span class="p">(</span><span class="n">infile</span><span class="p">,</span><span class="w"> </span><span class="n">header</span><span class="o">=</span><span class="nb">T</span><span class="p">,</span><span class="w"> </span><span class="n">sep</span><span class="o">=</span><span class="s2">"\t"</span><span class="p">)</span><span class="w">
</span><span class="n">head</span><span class="p">(</span><span class="n">normal_data</span><span class="p">)</span><span class="w">
</span><span class="n">head</span><span class="p">(</span><span class="n">tumor_data</span><span class="p">)</span><span class="w">

</span><span class="c1"># combine the two data sources</span><span class="w">
</span><span class="n">data</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">data.frame</span><span class="p">(</span><span class="n">chr</span><span class="o">=</span><span class="n">normal_data</span><span class="o">$</span><span class="n">chr</span><span class="p">,</span><span class="w"> </span><span class="n">posn</span><span class="o">=</span><span class="n">normal_data</span><span class="o">$</span><span class="n">posn</span><span class="p">,</span><span class="w"> </span><span class="n">normalDepth</span><span class="o">=</span><span class="n">normal_data</span><span class="o">$</span><span class="n">refCount</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="n">normal_data</span><span class="o">$</span><span class="n">nonRefCount</span><span class="p">,</span><span class="w"> </span><span class="n">tumorDepth</span><span class="o">=</span><span class="n">tumor_data</span><span class="o">$</span><span class="n">refCount</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="n">tumor_data</span><span class="o">$</span><span class="n">nonRefCount</span><span class="p">)</span><span class="w">
</span><span class="nf">dim</span><span class="p">(</span><span class="n">data</span><span class="p">)</span><span class="w">
</span><span class="c1"># filter by min_depth</span><span class="w">
</span><span class="n">data</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">subset</span><span class="p">(</span><span class="n">data</span><span class="p">,</span><span class="w"> </span><span class="n">normalDepth</span><span class="w"> </span><span class="o">&gt;=</span><span class="w"> </span><span class="n">min_depth</span><span class="w"> </span><span class="o">&amp;</span><span class="w"> </span><span class="n">tumorDepth</span><span class="w"> </span><span class="o">&gt;=</span><span class="w"> </span><span class="n">min_depth</span><span class="p">)</span><span class="w">
</span><span class="nf">dim</span><span class="p">(</span><span class="n">data</span><span class="p">)</span><span class="w">
</span><span class="c1"># compute the logR (without GC correction)</span><span class="w">
</span><span class="n">data</span><span class="o">$</span><span class="n">logR</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">log</span><span class="p">(</span><span class="n">data</span><span class="o">$</span><span class="n">tumorDepth</span><span class="o">/</span><span class="n">data</span><span class="o">$</span><span class="n">normalDepth</span><span class="p">,</span><span class="w"> </span><span class="m">2</span><span class="p">)</span><span class="w">
</span><span class="c1"># histogram of logR should resemble a Normal distribution</span><span class="w">
</span><span class="n">hist</span><span class="p">(</span><span class="n">data</span><span class="o">$</span><span class="n">logR</span><span class="p">,</span><span class="w"> </span><span class="n">breaks</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">30</span><span class="p">)</span><span class="w">
</span><span class="n">plot</span><span class="p">(</span><span class="m">1</span><span class="o">:</span><span class="nf">length</span><span class="p">(</span><span class="n">data</span><span class="o">$</span><span class="n">logR</span><span class="p">),</span><span class="w"> </span><span class="n">data</span><span class="o">$</span><span class="n">logR</span><span class="p">,</span><span class="w"> </span><span class="n">type</span><span class="o">=</span><span class="s1">'l'</span><span class="p">)</span><span class="w">

</span><span class="c1"># now, use Titan function loadAlleleCounts to form the data that is accepted by Titan functions</span><span class="w">
</span><span class="n">titan_data</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">loadAlleleCounts</span><span class="p">(</span><span class="n">tumor_data</span><span class="p">)</span><span class="w">
</span><span class="nf">dim</span><span class="p">(</span><span class="n">titan_data</span><span class="p">)</span><span class="w">
</span><span class="c1"># we want to trim this dataset to be the same length as data declared above</span><span class="w">
</span><span class="n">titan_data</span><span class="o">&lt;-</span><span class="n">subset</span><span class="p">(</span><span class="n">titan_data</span><span class="p">,</span><span class="w"> </span><span class="n">chr</span><span class="w"> </span><span class="o">%in%</span><span class="w"> </span><span class="n">data</span><span class="o">$</span><span class="n">chr</span><span class="w"> </span><span class="o">&amp;</span><span class="w"> </span><span class="n">posn</span><span class="w"> </span><span class="o">%in%</span><span class="w"> </span><span class="n">data</span><span class="o">$</span><span class="n">posn</span><span class="p">)</span><span class="w">
</span><span class="nf">dim</span><span class="p">(</span><span class="n">titan_data</span><span class="p">)</span><span class="w">
</span><span class="nf">dim</span><span class="p">(</span><span class="n">data</span><span class="p">)</span><span class="w">
</span><span class="c1"># now set the logR value for titan_data</span><span class="w">
</span><span class="n">titan_data</span><span class="o">$</span><span class="n">logR</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">data</span><span class="o">$</span><span class="n">logR</span><span class="w">
</span><span class="c1"># run the filterData function, which essentially turns X, Y into 23, 24</span><span class="w">
</span><span class="n">titan_data</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">filterData</span><span class="p">(</span><span class="n">titan_data</span><span class="p">,</span><span class="w"> </span><span class="m">1</span><span class="o">:</span><span class="m">24</span><span class="p">,</span><span class="w"> </span><span class="n">minDepth</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">min_depth</span><span class="p">,</span><span class="w"> </span><span class="n">maxDepth</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">max_depth</span><span class="p">,</span><span class="w"> </span><span class="n">map</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">NULL</span><span class="p">)</span><span class="w">
</span></code></pre></div></div>

<p>The inline comments describe the code in sufficient detail. But basically, I’m computing \(log_2(N^T/N^N)\), where \(N^T, N^N\) denote depth for tumour and normal samples respectively.</p>

<h4 id="notes-4">Notes</h4>
<ul>
  <li>I noticed that the tutorial code in the vignette is not using BAF data. I may have missed something but looking over the package documentation, it doesn’t seem like TitanCNA uses BAF data anywhere.</li>
</ul>

<h2 id="references">References</h2>
<p>[1] Ha G, Roth A, Khattra J, Ho J, Yap D, Prentice LM, Melnyk N, McPherson A, Bashashati A, Laks E, Biele J. TITAN: inference of copy number architectures in clonal cell populations from tumor whole-genome sequence data. Genome research. 2014 Jul 24:gr-180281.</p>

<p>[2] <a href="https://bioconductor.org/packages/release/bioc/vignettes/TitanCNA/inst/doc/TitanCNA.pdf">TitanCNA Vignette</a>.</p>

  </article>

</div>

      </div>
    </div>

    <footer class="site-footer">

  <div class="wrapper">

<!--    <h2 class="footer-heading">Seong-Hwan Jun</h2> -->

    <div class="footer-col-wrapper">
      <div class="footer-col  footer-col-3">
        <ul class="contact-list">
          <li>Address:</li>
          <li>265 Crittenden Blvd</li> 
          <li>Box 630</li>
          <li>Rochester, NY 14642</li>
        </ul>
      </div>

      <div class="footer-col  footer-col-3">
        <ul class="social-media-list">
          <!--<li>Email: seonghwan_jun AT urmc DOT rochester DOT edu</li>-->

          
          <li>
            <a href="https://github.com/junseonghwan">
              <span class="icon  icon--github">
                <svg viewBox="0 0 16 16">
                  <path fill="#828282" d="M7.999,0.431c-4.285,0-7.76,3.474-7.76,7.761 c0,3.428,2.223,6.337,5.307,7.363c0.388,0.071,0.53-0.168,0.53-0.374c0-0.184-0.007-0.672-0.01-1.32 c-2.159,0.469-2.614-1.04-2.614-1.04c-0.353-0.896-0.862-1.135-0.862-1.135c-0.705-0.481,0.053-0.472,0.053-0.472 c0.779,0.055,1.189,0.8,1.189,0.8c0.692,1.186,1.816,0.843,2.258,0.645c0.071-0.502,0.271-0.843,0.493-1.037 C4.86,11.425,3.049,10.76,3.049,7.786c0-0.847,0.302-1.54,0.799-2.082C3.768,5.507,3.501,4.718,3.924,3.65 c0,0,0.652-0.209,2.134,0.796C6.677,4.273,7.34,4.187,8,4.184c0.659,0.003,1.323,0.089,1.943,0.261 c1.482-1.004,2.132-0.796,2.132-0.796c0.423,1.068,0.157,1.857,0.077,2.054c0.497,0.542,0.798,1.235,0.798,2.082 c0,2.981-1.814,3.637-3.543,3.829c0.279,0.24,0.527,0.713,0.527,1.437c0,1.037-0.01,1.874-0.01,2.129 c0,0.208,0.14,0.449,0.534,0.373c3.081-1.028,5.302-3.935,5.302-7.362C15.76,3.906,12.285,0.431,7.999,0.431z"/>
                </svg>
              </span>

              <span class="username">junseonghwan</span>
            </a>
          </li>
          

          
        </ul>
      </div>
<!--
      <div class="footer-col  footer-col-3">
        <p class="text"></p>
      </div>
    </div>
-->
  </div>

</footer>


  </body>

</html>
