<!DOCTYPE html>
<html>

  <head>
  <meta charset="utf-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="viewport" content="width=device-width, initial-scale=1">

  <title>Ascat</title>
  <meta name="description" content="I use the pipeline developed at SciLifeLab called Sarek, which uses nextflow and performs the standard processing on NGS data as recommended by GATK best pra...">

  <link rel="stylesheet" href="/css/main.css">
  <link rel="stylesheet" href="/css/bootstrap.css">
  <link rel="canonical" href="junseonghwan.github.io/2019/07/08/ascat.html">
  <link rel="alternate" type="application/rss+xml" title="Seong-Hwan Jun" href="junseonghwan.github.io/feed.xml" />

  

  <script type="text/javascript" async
    src="https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-MML-AM_CHTML">
  </script>
  <!--
  <script src="https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML" type="text/javascript"></script>-->

</head>


  <body>

    <header class="site-header">

  <div class="wrapper">

    <a class="site-title" href="/">Seong-Hwan Jun</a>

    <nav class="site-nav">
      <a href="#" class="menu-icon">
        <svg viewBox="0 0 18 15">
          <path fill="#424242" d="M18,1.484c0,0.82-0.665,1.484-1.484,1.484H1.484C0.665,2.969,0,2.304,0,1.484l0,0C0,0.665,0.665,0,1.484,0 h15.031C17.335,0,18,0.665,18,1.484L18,1.484z"/>
          <path fill="#424242" d="M18,7.516C18,8.335,17.335,9,16.516,9H1.484C0.665,9,0,8.335,0,7.516l0,0c0-0.82,0.665-1.484,1.484-1.484 h15.031C17.335,6.031,18,6.696,18,7.516L18,7.516z"/>
          <path fill="#424242" d="M18,13.516C18,14.335,17.335,15,16.516,15H1.484C0.665,15,0,14.335,0,13.516l0,0 c0-0.82,0.665-1.484,1.484-1.484h15.031C17.335,12.031,18,12.696,18,13.516L18,13.516z"/>
        </svg>
      </a>

      <div class="trigger">
        
          
          <a class="page-link" href="/code/">Code</a>
          
        
          
        
          
        
          
          <a class="page-link" href="/research/">Research</a>
          
        
      </div>
    </nav>

  </div>

</header>


    <div class="page-content">
      <div class="wrapper">
        <div class="post">

  <header class="post-header">
    <h1 class="post-title">Ascat</h1>
    <p class="post-meta">Jul 8, 2019</p>
  </header>

  <article class="post-content">
    <p>I use the pipeline developed at SciLifeLab called <a href="https://github.com/SciLifeLab/Sarek">Sarek</a>, which uses nextflow and performs the standard processing on NGS data as recommended by <a href="https://software.broadinstitute.org/gatk/best-practices/">GATK best practices</a>. This produces BAM files for tumor and normal samples.</p>

<p>One of the important steps in analyzing tumor NGS data is copy number calling. Sarek ships with ascat and it provides details on how to use it on NGS data <a href="https://github.com/SciLifeLab/Sarek/blob/master/docs/ASCAT.md">here</a>. This post describes the steps that I took to run ascat on ALL data published with this <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4273416/">paper</a>.</p>

<h1 id="identifying-snp-positions">Identifying SNP positions</h1>
<p>It appears that ascat was originally developed for SNP array data. Ascat analyzes logR and BAF values at the positions probed using the SNP array. But the data that we will analyze is NGS data from tumor and normal samples so we need to compute these values and specify the positions that we will analyze.</p>

<p>Fortunately, this step has already been done for us by SciLifeLab team. A position is chosen if it is <a href="https://gatkforums.broadinstitute.org/gatk/discussion/6455/biallelic-vs-multiallelic-sites">bi-allelic</a> and if minor allele frequencies is greater than 0.3 (see <a href="https://github.com/SciLifeLab/Sarek/blob/master/docs/ASCAT.md#loci-file">here</a>).</p>

<h1 id="counting-a-and-b-alleles">Counting A and B alleles</h1>
<p>The next step is to count up the number of major and minor alleles at each of the identified positions. For this, we can use <a href="https://github.com/cancerit/alleleCount">alleleCount</a>. It is already installed on Uppmax</p>

<div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>module load bioinfo-tools alleleCount
alleleCounter -l [path_to_loci_file] -r [path_to_human_ref_genome] -b path_to_tumor.bam -o output.allecount
</code></pre></div></div>

<p>The two input files are the loci file, the SNP positions identified in the previous step and the path to reference human genome.</p>

<h1 id="ascat-si">Ascat SI</h1>
<p>I had to stare at Eq. S5 in the SI for quite some time: <script type="math/tex">b_i = \frac{1 - \rho + \rho n_B}{2 (1 - \rho) + \rho (n_A + n_B)}</script>. Here <script type="math/tex">\rho</script> refers to tumor fraction and <script type="math/tex">n_A, n_B</script> denote the true number of major and minor alleles. The observed data <script type="math/tex">\hat{b}_i = n_B/(n_A + n_B)</script> is a noisy version of the true value <script type="math/tex">b_i</script>.</p>

<p>This expression makes sense once we realize that this is an SNP position that is heterozygous (i.e., germline variant). Therefore, <script type="math/tex">(1-\rho)</script> comes from having one copy of the minor allele since this is the fraction that are healthy (one of major and one of minor copy). The other term, <script type="math/tex">\rho n_B</script> is more straight forward, as it refers to the number of copies of minor allele in the tumor population.</p>

  </article>

</div>

      </div>
    </div>

    <footer class="site-footer">

  <div class="wrapper">

<!--    <h2 class="footer-heading">Seong-Hwan Jun</h2> -->

    <div class="footer-col-wrapper">
      <div class="footer-col  footer-col-3">
        <ul class="contact-list">
          <li>Address:</li>
          <li>Seong-Hwan Jun</li>
          <li>1100 Fairview Ave. N S2-140</li>
          <li>Seattle, WA 98109</li> 
        </ul>
      </div>

      <div class="footer-col  footer-col-3">
        <ul class="social-media-list">
          <li>Email: sjun2 AT fredhutch dot org</li>

          
          <li>
            <a href="https://github.com/junseonghwan">
              <span class="icon  icon--github">
                <svg viewBox="0 0 16 16">
                  <path fill="#828282" d="M7.999,0.431c-4.285,0-7.76,3.474-7.76,7.761 c0,3.428,2.223,6.337,5.307,7.363c0.388,0.071,0.53-0.168,0.53-0.374c0-0.184-0.007-0.672-0.01-1.32 c-2.159,0.469-2.614-1.04-2.614-1.04c-0.353-0.896-0.862-1.135-0.862-1.135c-0.705-0.481,0.053-0.472,0.053-0.472 c0.779,0.055,1.189,0.8,1.189,0.8c0.692,1.186,1.816,0.843,2.258,0.645c0.071-0.502,0.271-0.843,0.493-1.037 C4.86,11.425,3.049,10.76,3.049,7.786c0-0.847,0.302-1.54,0.799-2.082C3.768,5.507,3.501,4.718,3.924,3.65 c0,0,0.652-0.209,2.134,0.796C6.677,4.273,7.34,4.187,8,4.184c0.659,0.003,1.323,0.089,1.943,0.261 c1.482-1.004,2.132-0.796,2.132-0.796c0.423,1.068,0.157,1.857,0.077,2.054c0.497,0.542,0.798,1.235,0.798,2.082 c0,2.981-1.814,3.637-3.543,3.829c0.279,0.24,0.527,0.713,0.527,1.437c0,1.037-0.01,1.874-0.01,2.129 c0,0.208,0.14,0.449,0.534,0.373c3.081-1.028,5.302-3.935,5.302-7.362C15.76,3.906,12.285,0.431,7.999,0.431z"/>
                </svg>
              </span>

              <span class="username">junseonghwan</span>
            </a>
          </li>
          

          
        </ul>
      </div>
<!--
      <div class="footer-col  footer-col-3">
        <p class="text"></p>
      </div>
    </div>
-->
  </div>

</footer>


  </body>

</html>
