<!DOCTYPE html>
<html>

  <head>
  <meta charset="utf-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="viewport" content="width=device-width, initial-scale=1">

  <title>PhylEx: Clonal tree reconstruction via integration of bulk DNA-seq and scRNA-seq data</title>
  <meta name="description" content="Revealing the evolutionary history of the cancer cells is challenging given the standard approach of single time point sequencing, as we cannot observe chang...">

  <link rel="stylesheet" href="/css/main.css">
  <link rel="stylesheet" href="/css/bootstrap.css">
  <link rel="canonical" href="junseonghwan.github.io/2021/01/29/phylex.html">
  <link rel="alternate" type="application/rss+xml" title="Seong-Hwan Jun" href="junseonghwan.github.io/feed.xml" />

  

  <script type="text/javascript" async
    src="https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-MML-AM_CHTML">
  </script>
  <!--
  <script src="https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML" type="text/javascript"></script>-->

</head>


  <body>

    <header class="site-header">

  <div class="wrapper">

    <a class="site-title" href="/">Seong-Hwan Jun</a>

    <nav class="site-nav">
      <a href="#" class="menu-icon">
        <svg viewBox="0 0 18 15">
          <path fill="#424242" d="M18,1.484c0,0.82-0.665,1.484-1.484,1.484H1.484C0.665,2.969,0,2.304,0,1.484l0,0C0,0.665,0.665,0,1.484,0 h15.031C17.335,0,18,0.665,18,1.484L18,1.484z"/>
          <path fill="#424242" d="M18,7.516C18,8.335,17.335,9,16.516,9H1.484C0.665,9,0,8.335,0,7.516l0,0c0-0.82,0.665-1.484,1.484-1.484 h15.031C17.335,6.031,18,6.696,18,7.516L18,7.516z"/>
          <path fill="#424242" d="M18,13.516C18,14.335,17.335,15,16.516,15H1.484C0.665,15,0,14.335,0,13.516l0,0 c0-0.82,0.665-1.484,1.484-1.484h15.031C17.335,12.031,18,12.696,18,13.516L18,13.516z"/>
        </svg>
      </a>

      <div class="trigger">
        
          
          <a class="page-link" href="/code/">Code</a>
          
        
          
        
          
        
          
          <a class="page-link" href="/research/">Research</a>
          
        
      </div>
    </nav>

  </div>

</header>


    <div class="page-content">
      <div class="wrapper">
        <div class="post">

  <header class="post-header">
    <h1 class="post-title">PhylEx: Clonal tree reconstruction via integration of bulk DNA-seq and scRNA-seq data</h1>
    <p class="post-meta">Jan 29, 2021</p>
  </header>

  <article class="post-content">
    <p>Revealing the evolutionary history of the cancer cells is challenging given the standard approach of single time point sequencing, as we cannot observe changes to cancer cells across time.
This challenge has not deterred researchers from tackling the problem: using acute statistical models, researchers have developed methods to reconstruct clonal trees using only the cells sequenced at a single time point.
In the early stages, the clone reconstruction approaches relied solely on the bulk DNA-seq data; however, with recent advances in single-cell technologies, it is now possible to resolve the ancestral structure with greater accuracy than ever before.</p>

<p>The single-cell transcriptomics (scRNA-seq) is gaining popularity as it can characterize cancer in its functional state.
This trend provides an opportunity to integrate scRNA-seq with bulk DNA-seq for clone tree reconstruction.
PhylEx is a Bayesian method for clone tree reconstruction on single nucleotide variants (SNVs) found in bulk DNA-seq with scRNA-seq data. 
Using the bulk data, we can identify SNVs and the variant allele frequencies of each of the SNVs with high confidence. 
From the scRNA-seq data, we can take advantage of the co-occurrence of SNVs in single cells to identify branching events with high accuracy.</p>

<p>We deployed PhylEx to analyze high-grade serous ovarian cancer (HGSOC) cell-line data that consists of a primary as well as a metastatic tumor [1].
We performed Smart-seq3, a sequencing method that performs full-length sequencing [2].</p>

<p><img src="/images/PhylEx.png" alt="Fig" style="left;margin-right: 30px;margin-top: 7px; height=&quot;400px" width="700px" />
<em>Figure 1. <strong>A</strong>. Ground truth tree labeled with clone names and the number of SNVs assigned to each clone. <strong>B</strong>. PhylEx reconstructed tree using bulk DNA-seq and scRNA-seq. The number of SNVs of each type is labeled. <strong>C</strong>. ZINB-WaVE dimensionality reduction on gene expression data with trajectory analysis results overlayed (arrows). <strong>D</strong>. ZINB-WaVE dimension plot with color-coding using Gaussian mixture clustering and with trajectory analysis result overlayed.</em></p>

<p>PhylEx showed high-concordance to the ground truth tree (Figure 1A-B).
The clone ABCD and its descendants are from the primary region, whereas clones EFGHI and EF are from the metastatic region in Figure 1A.
Aside from merging EFGHI and EF clones, PhylEx reconstructs the tree topology correctly.</p>

<p>We performed dimensionality reduction using the zero-inflated negative binomial model (ZINB-WaVE) on the raw expression counts.
Then, we performed pseudo-time trajectory analysis on the reduced dimension using Slingshot; to that end, we first assigned cells to clones by matching the cells’ observed genotypes to the genotypes of the clones based on the SNVs assigned to each clone.
As pseudo-time trajectory analysis attempts to reconstruct the clones’ temporal ordering from the single-cell gene expression data, it serves as a useful comparison.
We found that the trajectory inferred was generally incorrect (Figure 1C); in particular, cells assigned to clone CD were inferred as a <em>sibling</em> to ABCD clone. 
Furthermore, if model-based clustering was applied to label the cells before performing Slingshot, the cells were grouped arbitrarily based on expression profile. Hence, the trajectory analysis did not yield any meaningful results.
We have performed gene set enrichment analysis using the MSigDB Gene ontology set to compare the clones [3].</p>

<p>Our results show that cells in the metastatic region were found to down-regulate pathways related to the immune system (Table 1).
Together, these analysis demonstrate that reconstructing the evolutionary path taken by the cells can reveal insights that would otherwise be unavailable using only the gene expression data.</p>

<p><em>Table 1: Top 10 down-regulated pathways in the metastatic region compared to the primary region.</em></p>
<table class="table table-striped">
  <thead>
    <tr>
      <th scope="col">#</th>
      <th scope="col">Gene ontology</th>
      <th scope="col">P-value</th>
      <th scope="col">FDR</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th scope="row">1</th>
      <td>MHC protein complex</td>
      <td>4.85e-15</td>
      <td>1.32e-11</td>
    </tr>
    <tr>
      <th scope="row">2</th>
      <td>Antigen processing and presentation of endogenous antigen</td>
      <td>6.40e-15</td>
      <td>1.32e-11</td>
    </tr>
    <tr>
      <th scope="row">3</th>
      <td>MHC class I protein complex</td>
      <td>6.57e-14</td>
      <td>8.08e-11 </td>
    </tr>
    <tr>
      <th scope="row">4</th>
      <td>Response to type I interferon</td>
      <td>2.86e-11</td>
      <td>1.60e-08</td>
    </tr>
    <tr>
      <th scope="row">5</th>
      <td>Antigen processing and presentation of endogenous peptide antigen</td>
      <td>2.61e-10</td>
      <td>1.14e-07</td>
    </tr>
    <tr>
      <th scope="row">6</th>
      <td>Positive regulation of T cell mediated cytotoxicity</td>
      <td>7.96e-09</td>
      <td>2.51e-06</td>
    </tr>
    <tr>
      <th scope="row">7</th>
      <td>Interferon Gamma mediated signaling pathway</td>
      <td>2.08e-08</td>
      <td>5.72e-06</td>
    </tr>
    <tr>
      <th scope="row">8</th>
      <td>Regulation of T cell mediated cytotoxicity</td>
      <td>2.37e-08</td>
      <td>6.07e-06</td>
    </tr>
    <tr>
      <th scope="row">9</th>
      <td>Positive regulation of antigen processing and presentation</td>
      <td>4.99e-07</td>
      <td>9.59e-05</td>
    </tr>
    <tr>
      <th scope="row">10</th>
      <td>Detection of other organism</td>
      <td>6.87e-07</td>
      <td>1.28e-04</td>
    </tr>    
  </tbody>
</table>

<p>[1] Laks E, McPherson A, Zahn H, Lai D, Steif A, Brimhall J, Biele J, Wang B, Masud T, Ting J, Grewal D, Nielsen C, Leung S, Bojilova V, Smith M, Golovko O, Poon S, Eirew P, Kabeer F, Ruiz de Algara T, Lee SR, Taghiyar MJ, Huebner C, Ngo J, Chan T, Vatrt-Watts S, Walters P, Abrar N, Chan S, Wiens M, Martin L, Scott RW, Underhill TM, Chavez E, Steidl C, Da Costa D, Ma Y, Coope RJN, Corbett R, Pleasance S, Moore R, Mungall AJ, Mar C, Cafferty F, Gelmon K, Chia S; CRUK IMAXT Grand Challenge Team, Marra MA, Hansen C, Shah SP, Aparicio S. Clonal Decomposition and DNA Replication States Defined by Scaled Single-Cell Genome Sequencing. Cell. 2019 Nov 14;179(5):1207-1221.e22. doi: 10.1016/j.cell.2019.10.026. PMID: 31730858; PMCID: PMC6912164.</p>

<p>[2] Hagemann-Jensen, M., Ziegenhain, C., Chen, P. et al. Single-cell RNA counting at allele and isoform resolution using Smart-seq3. Nat Biotechnol 38, 708–714 (2020). https://doi.org/10.1038/s41587-020-0497-0</p>

<p>[3] Arthur Liberzon, Aravind Subramanian, Reid Pinchback, Helga Thorvaldsdóttir, Pablo Tamayo, Jill P. Mesirov, Molecular signatures database (MSigDB) 3.0, Bioinformatics, Volume 27, Issue 12, 15 June 2011, Pages 1739–1740, https://doi.org/10.1093/bioinformatics/btr260</p>

  </article>

</div>

      </div>
    </div>

    <footer class="site-footer">

  <div class="wrapper">

<!--    <h2 class="footer-heading">Seong-Hwan Jun</h2> -->

    <div class="footer-col-wrapper">
      <div class="footer-col  footer-col-3">
        <ul class="contact-list">
          <li>Address:</li>
          <li>265 Crittenden Blvd</li> 
          <li>Box 630</li>
          <li>Rochester, NY 14642</li>
        </ul>
      </div>

      <div class="footer-col  footer-col-3">
        <ul class="social-media-list">
          <!--<li>Email: seonghwan_jun AT urmc DOT rochester DOT edu</li>-->

          
          <li>
            <a href="https://github.com/junseonghwan">
              <span class="icon  icon--github">
                <svg viewBox="0 0 16 16">
                  <path fill="#828282" d="M7.999,0.431c-4.285,0-7.76,3.474-7.76,7.761 c0,3.428,2.223,6.337,5.307,7.363c0.388,0.071,0.53-0.168,0.53-0.374c0-0.184-0.007-0.672-0.01-1.32 c-2.159,0.469-2.614-1.04-2.614-1.04c-0.353-0.896-0.862-1.135-0.862-1.135c-0.705-0.481,0.053-0.472,0.053-0.472 c0.779,0.055,1.189,0.8,1.189,0.8c0.692,1.186,1.816,0.843,2.258,0.645c0.071-0.502,0.271-0.843,0.493-1.037 C4.86,11.425,3.049,10.76,3.049,7.786c0-0.847,0.302-1.54,0.799-2.082C3.768,5.507,3.501,4.718,3.924,3.65 c0,0,0.652-0.209,2.134,0.796C6.677,4.273,7.34,4.187,8,4.184c0.659,0.003,1.323,0.089,1.943,0.261 c1.482-1.004,2.132-0.796,2.132-0.796c0.423,1.068,0.157,1.857,0.077,2.054c0.497,0.542,0.798,1.235,0.798,2.082 c0,2.981-1.814,3.637-3.543,3.829c0.279,0.24,0.527,0.713,0.527,1.437c0,1.037-0.01,1.874-0.01,2.129 c0,0.208,0.14,0.449,0.534,0.373c3.081-1.028,5.302-3.935,5.302-7.362C15.76,3.906,12.285,0.431,7.999,0.431z"/>
                </svg>
              </span>

              <span class="username">junseonghwan</span>
            </a>
          </li>
          

          
        </ul>
      </div>
<!--
      <div class="footer-col  footer-col-3">
        <p class="text"></p>
      </div>
    </div>
-->
  </div>

</footer>


  </body>

</html>
